// @martifylabs/mesh-react/dist/mesh-react.js
import Le, { useRef as yr, useDebugValue as kt, useContext as _e, createElement as wr, createContext as br, useState as de, useCallback as ct, useEffect as He } from "react";
import { BrowserWallet as Ut, Transaction as Sr } from "@martifylabs/mesh";
var Ge = { exports: {} }, Y = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Tt;
function Er() {
  if (Tt)
    return Y;
  Tt = 1;
  var e = typeof Symbol == "function" && Symbol.for, n = e ? Symbol.for("react.element") : 60103, r = e ? Symbol.for("react.portal") : 60106, o = e ? Symbol.for("react.fragment") : 60107, i = e ? Symbol.for("react.strict_mode") : 60108, s = e ? Symbol.for("react.profiler") : 60114, c = e ? Symbol.for("react.provider") : 60109, p = e ? Symbol.for("react.context") : 60110, h = e ? Symbol.for("react.async_mode") : 60111, b = e ? Symbol.for("react.concurrent_mode") : 60111, g = e ? Symbol.for("react.forward_ref") : 60112, E = e ? Symbol.for("react.suspense") : 60113, L = e ? Symbol.for("react.suspense_list") : 60120, F = e ? Symbol.for("react.memo") : 60115, x = e ? Symbol.for("react.lazy") : 60116, I = e ? Symbol.for("react.block") : 60121, k = e ? Symbol.for("react.fundamental") : 60117, N = e ? Symbol.for("react.responder") : 60118, S = e ? Symbol.for("react.scope") : 60119;
  function T(d) {
    if (typeof d == "object" && d !== null) {
      var oe = d.$$typeof;
      switch (oe) {
        case n:
          switch (d = d.type, d) {
            case h:
            case b:
            case o:
            case s:
            case i:
            case E:
              return d;
            default:
              switch (d = d && d.$$typeof, d) {
                case p:
                case g:
                case x:
                case F:
                case c:
                  return d;
                default:
                  return oe;
              }
          }
        case r:
          return oe;
      }
    }
  }
  function C(d) {
    return T(d) === b;
  }
  return Y.AsyncMode = h, Y.ConcurrentMode = b, Y.ContextConsumer = p, Y.ContextProvider = c, Y.Element = n, Y.ForwardRef = g, Y.Fragment = o, Y.Lazy = x, Y.Memo = F, Y.Portal = r, Y.Profiler = s, Y.StrictMode = i, Y.Suspense = E, Y.isAsyncMode = function(d) {
    return C(d) || T(d) === h;
  }, Y.isConcurrentMode = C, Y.isContextConsumer = function(d) {
    return T(d) === p;
  }, Y.isContextProvider = function(d) {
    return T(d) === c;
  }, Y.isElement = function(d) {
    return typeof d == "object" && d !== null && d.$$typeof === n;
  }, Y.isForwardRef = function(d) {
    return T(d) === g;
  }, Y.isFragment = function(d) {
    return T(d) === o;
  }, Y.isLazy = function(d) {
    return T(d) === x;
  }, Y.isMemo = function(d) {
    return T(d) === F;
  }, Y.isPortal = function(d) {
    return T(d) === r;
  }, Y.isProfiler = function(d) {
    return T(d) === s;
  }, Y.isStrictMode = function(d) {
    return T(d) === i;
  }, Y.isSuspense = function(d) {
    return T(d) === E;
  }, Y.isValidElementType = function(d) {
    return typeof d == "string" || typeof d == "function" || d === o || d === b || d === s || d === i || d === E || d === L || typeof d == "object" && d !== null && (d.$$typeof === x || d.$$typeof === F || d.$$typeof === c || d.$$typeof === p || d.$$typeof === g || d.$$typeof === k || d.$$typeof === N || d.$$typeof === S || d.$$typeof === I);
  }, Y.typeOf = T, Y;
}
var V = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Pt;
function Cr() {
  return Pt || (Pt = 1, process.env.NODE_ENV !== "production" && function() {
    var e = typeof Symbol == "function" && Symbol.for, n = e ? Symbol.for("react.element") : 60103, r = e ? Symbol.for("react.portal") : 60106, o = e ? Symbol.for("react.fragment") : 60107, i = e ? Symbol.for("react.strict_mode") : 60108, s = e ? Symbol.for("react.profiler") : 60114, c = e ? Symbol.for("react.provider") : 60109, p = e ? Symbol.for("react.context") : 60110, h = e ? Symbol.for("react.async_mode") : 60111, b = e ? Symbol.for("react.concurrent_mode") : 60111, g = e ? Symbol.for("react.forward_ref") : 60112, E = e ? Symbol.for("react.suspense") : 60113, L = e ? Symbol.for("react.suspense_list") : 60120, F = e ? Symbol.for("react.memo") : 60115, x = e ? Symbol.for("react.lazy") : 60116, I = e ? Symbol.for("react.block") : 60121, k = e ? Symbol.for("react.fundamental") : 60117, N = e ? Symbol.for("react.responder") : 60118, S = e ? Symbol.for("react.scope") : 60119;
    function T(l) {
      return typeof l == "string" || typeof l == "function" || l === o || l === b || l === s || l === i || l === E || l === L || typeof l == "object" && l !== null && (l.$$typeof === x || l.$$typeof === F || l.$$typeof === c || l.$$typeof === p || l.$$typeof === g || l.$$typeof === k || l.$$typeof === N || l.$$typeof === S || l.$$typeof === I);
    }
    function C(l) {
      if (typeof l == "object" && l !== null) {
        var q = l.$$typeof;
        switch (q) {
          case n:
            var Z = l.type;
            switch (Z) {
              case h:
              case b:
              case o:
              case s:
              case i:
              case E:
                return Z;
              default:
                var P = Z && Z.$$typeof;
                switch (P) {
                  case p:
                  case g:
                  case x:
                  case F:
                  case c:
                    return P;
                  default:
                    return q;
                }
            }
          case r:
            return q;
        }
      }
    }
    var d = h, oe = b, pe = p, se = c, le = n, he = g, ye = o, ae = x, H = F, ee = r, K = s, we = i, ce = E, ie = !1;
    function be(l) {
      return ie || (ie = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), me(l) || C(l) === h;
    }
    function me(l) {
      return C(l) === b;
    }
    function Se(l) {
      return C(l) === p;
    }
    function w(l) {
      return C(l) === c;
    }
    function v(l) {
      return typeof l == "object" && l !== null && l.$$typeof === n;
    }
    function y(l) {
      return C(l) === g;
    }
    function A(l) {
      return C(l) === o;
    }
    function a(l) {
      return C(l) === x;
    }
    function D(l) {
      return C(l) === F;
    }
    function f(l) {
      return C(l) === r;
    }
    function B(l) {
      return C(l) === s;
    }
    function O(l) {
      return C(l) === i;
    }
    function j(l) {
      return C(l) === E;
    }
    V.AsyncMode = d, V.ConcurrentMode = oe, V.ContextConsumer = pe, V.ContextProvider = se, V.Element = le, V.ForwardRef = he, V.Fragment = ye, V.Lazy = ae, V.Memo = H, V.Portal = ee, V.Profiler = K, V.StrictMode = we, V.Suspense = ce, V.isAsyncMode = be, V.isConcurrentMode = me, V.isContextConsumer = Se, V.isContextProvider = w, V.isElement = v, V.isForwardRef = y, V.isFragment = A, V.isLazy = a, V.isMemo = D, V.isPortal = f, V.isProfiler = B, V.isStrictMode = O, V.isSuspense = j, V.isValidElementType = T, V.typeOf = C;
  }()), V;
}
(function(e) {
  process.env.NODE_ENV === "production" ? e.exports = Er() : e.exports = Cr();
})(Ge);
function _r(e) {
  function n(w, v, y, A, a) {
    for (var D = 0, f = 0, B = 0, O = 0, j, l, q = 0, Z = 0, P, J = P = j = 0, $ = 0, te = 0, Ee = 0, ne = 0, Ne = y.length, Re = Ne - 1, fe, _ = "", G = "", De = "", xe = "", Ce; $ < Ne; ) {
      if (l = y.charCodeAt($), $ === Re && f + O + B + D !== 0 && (f !== 0 && (l = f === 47 ? 10 : 47), O = B = D = 0, Ne++, Re++), f + O + B + D === 0) {
        if ($ === Re && (0 < te && (_ = _.replace(L, "")), 0 < _.trim().length)) {
          switch (l) {
            case 32:
            case 9:
            case 59:
            case 13:
            case 10:
              break;
            default:
              _ += y.charAt($);
          }
          l = 59;
        }
        switch (l) {
          case 123:
            for (_ = _.trim(), j = _.charCodeAt(0), P = 1, ne = ++$; $ < Ne; ) {
              switch (l = y.charCodeAt($)) {
                case 123:
                  P++;
                  break;
                case 125:
                  P--;
                  break;
                case 47:
                  switch (l = y.charCodeAt($ + 1)) {
                    case 42:
                    case 47:
                      e: {
                        for (J = $ + 1; J < Re; ++J)
                          switch (y.charCodeAt(J)) {
                            case 47:
                              if (l === 42 && y.charCodeAt(J - 1) === 42 && $ + 2 !== J) {
                                $ = J + 1;
                                break e;
                              }
                              break;
                            case 10:
                              if (l === 47) {
                                $ = J + 1;
                                break e;
                              }
                          }
                        $ = J;
                      }
                  }
                  break;
                case 91:
                  l++;
                case 40:
                  l++;
                case 34:
                case 39:
                  for (; $++ < Re && y.charCodeAt($) !== l; )
                    ;
              }
              if (P === 0)
                break;
              $++;
            }
            switch (P = y.substring(ne, $), j === 0 && (j = (_ = _.replace(E, "").trim()).charCodeAt(0)), j) {
              case 64:
                switch (0 < te && (_ = _.replace(L, "")), l = _.charCodeAt(1), l) {
                  case 100:
                  case 109:
                  case 115:
                  case 45:
                    te = v;
                    break;
                  default:
                    te = we;
                }
                if (P = n(v, te, P, l, a + 1), ne = P.length, 0 < ie && (te = r(we, _, Ee), Ce = p(3, P, te, v, H, ae, ne, l, a, A), _ = te.join(""), Ce !== void 0 && (ne = (P = Ce.trim()).length) === 0 && (l = 0, P = "")), 0 < ne)
                  switch (l) {
                    case 115:
                      _ = _.replace(oe, c);
                    case 100:
                    case 109:
                    case 45:
                      P = _ + "{" + P + "}";
                      break;
                    case 107:
                      _ = _.replace(S, "$1 $2"), P = _ + "{" + P + "}", P = K === 1 || K === 2 && s("@" + P, 3) ? "@-webkit-" + P + "@" + P : "@" + P;
                      break;
                    default:
                      P = _ + P, A === 112 && (P = (G += P, ""));
                  }
                else
                  P = "";
                break;
              default:
                P = n(v, r(v, _, Ee), P, A, a + 1);
            }
            De += P, P = Ee = te = J = j = 0, _ = "", l = y.charCodeAt(++$);
            break;
          case 125:
          case 59:
            if (_ = (0 < te ? _.replace(L, "") : _).trim(), 1 < (ne = _.length))
              switch (J === 0 && (j = _.charCodeAt(0), j === 45 || 96 < j && 123 > j) && (ne = (_ = _.replace(" ", ":")).length), 0 < ie && (Ce = p(1, _, v, w, H, ae, G.length, A, a, A)) !== void 0 && (ne = (_ = Ce.trim()).length) === 0 && (_ = "\0\0"), j = _.charCodeAt(0), l = _.charCodeAt(1), j) {
                case 0:
                  break;
                case 64:
                  if (l === 105 || l === 99) {
                    xe += _ + y.charAt($);
                    break;
                  }
                default:
                  _.charCodeAt(ne - 1) !== 58 && (G += i(_, j, l, _.charCodeAt(2)));
              }
            Ee = te = J = j = 0, _ = "", l = y.charCodeAt(++$);
        }
      }
      switch (l) {
        case 13:
        case 10:
          f === 47 ? f = 0 : 1 + j === 0 && A !== 107 && 0 < _.length && (te = 1, _ += "\0"), 0 < ie * me && p(0, _, v, w, H, ae, G.length, A, a, A), ae = 1, H++;
          break;
        case 59:
        case 125:
          if (f + O + B + D === 0) {
            ae++;
            break;
          }
        default:
          switch (ae++, fe = y.charAt($), l) {
            case 9:
            case 32:
              if (O + D + f === 0)
                switch (q) {
                  case 44:
                  case 58:
                  case 9:
                  case 32:
                    fe = "";
                    break;
                  default:
                    l !== 32 && (fe = " ");
                }
              break;
            case 0:
              fe = "\\0";
              break;
            case 12:
              fe = "\\f";
              break;
            case 11:
              fe = "\\v";
              break;
            case 38:
              O + f + D === 0 && (te = Ee = 1, fe = "\f" + fe);
              break;
            case 108:
              if (O + f + D + ee === 0 && 0 < J)
                switch ($ - J) {
                  case 2:
                    q === 112 && y.charCodeAt($ - 3) === 58 && (ee = q);
                  case 8:
                    Z === 111 && (ee = Z);
                }
              break;
            case 58:
              O + f + D === 0 && (J = $);
              break;
            case 44:
              f + B + O + D === 0 && (te = 1, fe += "\r");
              break;
            case 34:
            case 39:
              f === 0 && (O = O === l ? 0 : O === 0 ? l : O);
              break;
            case 91:
              O + f + B === 0 && D++;
              break;
            case 93:
              O + f + B === 0 && D--;
              break;
            case 41:
              O + f + D === 0 && B--;
              break;
            case 40:
              if (O + f + D === 0) {
                if (j === 0)
                  switch (2 * q + 3 * Z) {
                    case 533:
                      break;
                    default:
                      j = 1;
                  }
                B++;
              }
              break;
            case 64:
              f + B + O + D + J + P === 0 && (P = 1);
              break;
            case 42:
            case 47:
              if (!(0 < O + D + B))
                switch (f) {
                  case 0:
                    switch (2 * l + 3 * y.charCodeAt($ + 1)) {
                      case 235:
                        f = 47;
                        break;
                      case 220:
                        ne = $, f = 42;
                    }
                    break;
                  case 42:
                    l === 47 && q === 42 && ne + 2 !== $ && (y.charCodeAt(ne + 2) === 33 && (G += y.substring(ne, $ + 1)), fe = "", f = 0);
                }
          }
          f === 0 && (_ += fe);
      }
      Z = q, q = l, $++;
    }
    if (ne = G.length, 0 < ne) {
      if (te = v, 0 < ie && (Ce = p(2, G, te, w, H, ae, ne, A, a, A), Ce !== void 0 && (G = Ce).length === 0))
        return xe + G + De;
      if (G = te.join(",") + "{" + G + "}", K * ee !== 0) {
        switch (K !== 2 || s(G, 2) || (ee = 0), ee) {
          case 111:
            G = G.replace(C, ":-moz-$1") + G;
            break;
          case 112:
            G = G.replace(T, "::-webkit-input-$1") + G.replace(T, "::-moz-$1") + G.replace(T, ":-ms-input-$1") + G;
        }
        ee = 0;
      }
    }
    return xe + G + De;
  }
  function r(w, v, y) {
    var A = v.trim().split(k);
    v = A;
    var a = A.length, D = w.length;
    switch (D) {
      case 0:
      case 1:
        var f = 0;
        for (w = D === 0 ? "" : w[0] + " "; f < a; ++f)
          v[f] = o(w, v[f], y).trim();
        break;
      default:
        var B = f = 0;
        for (v = []; f < a; ++f)
          for (var O = 0; O < D; ++O)
            v[B++] = o(w[O] + " ", A[f], y).trim();
    }
    return v;
  }
  function o(w, v, y) {
    var A = v.charCodeAt(0);
    switch (33 > A && (A = (v = v.trim()).charCodeAt(0)), A) {
      case 38:
        return v.replace(N, "$1" + w.trim());
      case 58:
        return w.trim() + v.replace(N, "$1" + w.trim());
      default:
        if (0 < 1 * y && 0 < v.indexOf("\f"))
          return v.replace(N, (w.charCodeAt(0) === 58 ? "" : "$1") + w.trim());
    }
    return w + v;
  }
  function i(w, v, y, A) {
    var a = w + ";", D = 2 * v + 3 * y + 4 * A;
    if (D === 944) {
      w = a.indexOf(":", 9) + 1;
      var f = a.substring(w, a.length - 1).trim();
      return f = a.substring(0, w).trim() + f + ";", K === 1 || K === 2 && s(f, 1) ? "-webkit-" + f + f : f;
    }
    if (K === 0 || K === 2 && !s(a, 1))
      return a;
    switch (D) {
      case 1015:
        return a.charCodeAt(10) === 97 ? "-webkit-" + a + a : a;
      case 951:
        return a.charCodeAt(3) === 116 ? "-webkit-" + a + a : a;
      case 963:
        return a.charCodeAt(5) === 110 ? "-webkit-" + a + a : a;
      case 1009:
        if (a.charCodeAt(4) !== 100)
          break;
      case 969:
      case 942:
        return "-webkit-" + a + a;
      case 978:
        return "-webkit-" + a + "-moz-" + a + a;
      case 1019:
      case 983:
        return "-webkit-" + a + "-moz-" + a + "-ms-" + a + a;
      case 883:
        if (a.charCodeAt(8) === 45)
          return "-webkit-" + a + a;
        if (0 < a.indexOf("image-set(", 11))
          return a.replace(ye, "$1-webkit-$2") + a;
        break;
      case 932:
        if (a.charCodeAt(4) === 45)
          switch (a.charCodeAt(5)) {
            case 103:
              return "-webkit-box-" + a.replace("-grow", "") + "-webkit-" + a + "-ms-" + a.replace("grow", "positive") + a;
            case 115:
              return "-webkit-" + a + "-ms-" + a.replace("shrink", "negative") + a;
            case 98:
              return "-webkit-" + a + "-ms-" + a.replace("basis", "preferred-size") + a;
          }
        return "-webkit-" + a + "-ms-" + a + a;
      case 964:
        return "-webkit-" + a + "-ms-flex-" + a + a;
      case 1023:
        if (a.charCodeAt(8) !== 99)
          break;
        return f = a.substring(a.indexOf(":", 15)).replace("flex-", "").replace("space-between", "justify"), "-webkit-box-pack" + f + "-webkit-" + a + "-ms-flex-pack" + f + a;
      case 1005:
        return x.test(a) ? a.replace(F, ":-webkit-") + a.replace(F, ":-moz-") + a : a;
      case 1e3:
        switch (f = a.substring(13).trim(), v = f.indexOf("-") + 1, f.charCodeAt(0) + f.charCodeAt(v)) {
          case 226:
            f = a.replace(d, "tb");
            break;
          case 232:
            f = a.replace(d, "tb-rl");
            break;
          case 220:
            f = a.replace(d, "lr");
            break;
          default:
            return a;
        }
        return "-webkit-" + a + "-ms-" + f + a;
      case 1017:
        if (a.indexOf("sticky", 9) === -1)
          break;
      case 975:
        switch (v = (a = w).length - 10, f = (a.charCodeAt(v) === 33 ? a.substring(0, v) : a).substring(w.indexOf(":", 7) + 1).trim(), D = f.charCodeAt(0) + (f.charCodeAt(7) | 0)) {
          case 203:
            if (111 > f.charCodeAt(8))
              break;
          case 115:
            a = a.replace(f, "-webkit-" + f) + ";" + a;
            break;
          case 207:
          case 102:
            a = a.replace(f, "-webkit-" + (102 < D ? "inline-" : "") + "box") + ";" + a.replace(f, "-webkit-" + f) + ";" + a.replace(f, "-ms-" + f + "box") + ";" + a;
        }
        return a + ";";
      case 938:
        if (a.charCodeAt(5) === 45)
          switch (a.charCodeAt(6)) {
            case 105:
              return f = a.replace("-items", ""), "-webkit-" + a + "-webkit-box-" + f + "-ms-flex-" + f + a;
            case 115:
              return "-webkit-" + a + "-ms-flex-item-" + a.replace(se, "") + a;
            default:
              return "-webkit-" + a + "-ms-flex-line-pack" + a.replace("align-content", "").replace(se, "") + a;
          }
        break;
      case 973:
      case 989:
        if (a.charCodeAt(3) !== 45 || a.charCodeAt(4) === 122)
          break;
      case 931:
      case 953:
        if (he.test(w) === !0)
          return (f = w.substring(w.indexOf(":") + 1)).charCodeAt(0) === 115 ? i(w.replace("stretch", "fill-available"), v, y, A).replace(":fill-available", ":stretch") : a.replace(f, "-webkit-" + f) + a.replace(f, "-moz-" + f.replace("fill-", "")) + a;
        break;
      case 962:
        if (a = "-webkit-" + a + (a.charCodeAt(5) === 102 ? "-ms-" + a : "") + a, y + A === 211 && a.charCodeAt(13) === 105 && 0 < a.indexOf("transform", 10))
          return a.substring(0, a.indexOf(";", 27) + 1).replace(I, "$1-webkit-$2") + a;
    }
    return a;
  }
  function s(w, v) {
    var y = w.indexOf(v === 1 ? ":" : "{"), A = w.substring(0, v !== 3 ? y : 10);
    return y = w.substring(y + 1, w.length - 1), be(v !== 2 ? A : A.replace(le, "$1"), y, v);
  }
  function c(w, v) {
    var y = i(v, v.charCodeAt(0), v.charCodeAt(1), v.charCodeAt(2));
    return y !== v + ";" ? y.replace(pe, " or ($1)").substring(4) : "(" + v + ")";
  }
  function p(w, v, y, A, a, D, f, B, O, j) {
    for (var l = 0, q = v, Z; l < ie; ++l)
      switch (Z = ce[l].call(g, w, q, y, A, a, D, f, B, O, j)) {
        case void 0:
        case !1:
        case !0:
        case null:
          break;
        default:
          q = Z;
      }
    if (q !== v)
      return q;
  }
  function h(w) {
    switch (w) {
      case void 0:
      case null:
        ie = ce.length = 0;
        break;
      default:
        if (typeof w == "function")
          ce[ie++] = w;
        else if (typeof w == "object")
          for (var v = 0, y = w.length; v < y; ++v)
            h(w[v]);
        else
          me = !!w | 0;
    }
    return h;
  }
  function b(w) {
    return w = w.prefix, w !== void 0 && (be = null, w ? typeof w != "function" ? K = 1 : (K = 2, be = w) : K = 0), b;
  }
  function g(w, v) {
    var y = w;
    if (33 > y.charCodeAt(0) && (y = y.trim()), Se = y, y = [Se], 0 < ie) {
      var A = p(-1, v, y, y, H, ae, 0, 0, 0, 0);
      A !== void 0 && typeof A == "string" && (v = A);
    }
    var a = n(we, y, v, 0, 0);
    return 0 < ie && (A = p(-2, a, y, y, H, ae, a.length, 0, 0, 0), A !== void 0 && (a = A)), Se = "", ee = 0, ae = H = 1, a;
  }
  var E = /^\0+/g, L = /[\0\r\f]/g, F = /: */g, x = /zoo|gra/, I = /([,: ])(transform)/g, k = /,\r+?/g, N = /([\t\r\n ])*\f?&/g, S = /@(k\w+)\s*(\S*)\s*/, T = /::(place)/g, C = /:(read-only)/g, d = /[svh]\w+-[tblr]{2}/, oe = /\(\s*(.*)\s*\)/g, pe = /([\s\S]*?);/g, se = /-self|flex-/g, le = /[^]*?(:[rp][el]a[\w-]+)[^]*/, he = /stretch|:\s*\w+\-(?:conte|avail)/, ye = /([^-])(image-set\()/, ae = 1, H = 1, ee = 0, K = 1, we = [], ce = [], ie = 0, be = null, me = 0, Se = "";
  return g.use = h, g.set = b, e !== void 0 && b(e), g;
}
var Rr = {
  animationIterationCount: 1,
  borderImageOutset: 1,
  borderImageSlice: 1,
  borderImageWidth: 1,
  boxFlex: 1,
  boxFlexGroup: 1,
  boxOrdinalGroup: 1,
  columnCount: 1,
  columns: 1,
  flex: 1,
  flexGrow: 1,
  flexPositive: 1,
  flexShrink: 1,
  flexNegative: 1,
  flexOrder: 1,
  gridRow: 1,
  gridRowEnd: 1,
  gridRowSpan: 1,
  gridRowStart: 1,
  gridColumn: 1,
  gridColumnEnd: 1,
  gridColumnSpan: 1,
  gridColumnStart: 1,
  msGridRow: 1,
  msGridRowSpan: 1,
  msGridColumn: 1,
  msGridColumnSpan: 1,
  fontWeight: 1,
  lineHeight: 1,
  opacity: 1,
  order: 1,
  orphans: 1,
  tabSize: 1,
  widows: 1,
  zIndex: 1,
  zoom: 1,
  WebkitLineClamp: 1,
  fillOpacity: 1,
  floodOpacity: 1,
  stopOpacity: 1,
  strokeDasharray: 1,
  strokeDashoffset: 1,
  strokeMiterlimit: 1,
  strokeOpacity: 1,
  strokeWidth: 1
};
function xr(e) {
  var n = /* @__PURE__ */ Object.create(null);
  return function(r) {
    return n[r] === void 0 && (n[r] = e(r)), n[r];
  };
}
var Ar = /^((children|dangerouslySetInnerHTML|key|ref|autoFocus|defaultValue|defaultChecked|innerHTML|suppressContentEditableWarning|suppressHydrationWarning|valueLink|abbr|accept|acceptCharset|accessKey|action|allow|allowUserMedia|allowPaymentRequest|allowFullScreen|allowTransparency|alt|async|autoComplete|autoPlay|capture|cellPadding|cellSpacing|challenge|charSet|checked|cite|classID|className|cols|colSpan|content|contentEditable|contextMenu|controls|controlsList|coords|crossOrigin|data|dateTime|decoding|default|defer|dir|disabled|disablePictureInPicture|download|draggable|encType|enterKeyHint|form|formAction|formEncType|formMethod|formNoValidate|formTarget|frameBorder|headers|height|hidden|high|href|hrefLang|htmlFor|httpEquiv|id|inputMode|integrity|is|keyParams|keyType|kind|label|lang|list|loading|loop|low|marginHeight|marginWidth|max|maxLength|media|mediaGroup|method|min|minLength|multiple|muted|name|nonce|noValidate|open|optimum|pattern|placeholder|playsInline|poster|preload|profile|radioGroup|readOnly|referrerPolicy|rel|required|reversed|role|rows|rowSpan|sandbox|scope|scoped|scrolling|seamless|selected|shape|size|sizes|slot|span|spellCheck|src|srcDoc|srcLang|srcSet|start|step|style|summary|tabIndex|target|title|translate|type|useMap|value|width|wmode|wrap|about|datatype|inlist|prefix|property|resource|typeof|vocab|autoCapitalize|autoCorrect|autoSave|color|incremental|fallback|inert|itemProp|itemScope|itemType|itemID|itemRef|on|option|results|security|unselectable|accentHeight|accumulate|additive|alignmentBaseline|allowReorder|alphabetic|amplitude|arabicForm|ascent|attributeName|attributeType|autoReverse|azimuth|baseFrequency|baselineShift|baseProfile|bbox|begin|bias|by|calcMode|capHeight|clip|clipPathUnits|clipPath|clipRule|colorInterpolation|colorInterpolationFilters|colorProfile|colorRendering|contentScriptType|contentStyleType|cursor|cx|cy|d|decelerate|descent|diffuseConstant|direction|display|divisor|dominantBaseline|dur|dx|dy|edgeMode|elevation|enableBackground|end|exponent|externalResourcesRequired|fill|fillOpacity|fillRule|filter|filterRes|filterUnits|floodColor|floodOpacity|focusable|fontFamily|fontSize|fontSizeAdjust|fontStretch|fontStyle|fontVariant|fontWeight|format|from|fr|fx|fy|g1|g2|glyphName|glyphOrientationHorizontal|glyphOrientationVertical|glyphRef|gradientTransform|gradientUnits|hanging|horizAdvX|horizOriginX|ideographic|imageRendering|in|in2|intercept|k|k1|k2|k3|k4|kernelMatrix|kernelUnitLength|kerning|keyPoints|keySplines|keyTimes|lengthAdjust|letterSpacing|lightingColor|limitingConeAngle|local|markerEnd|markerMid|markerStart|markerHeight|markerUnits|markerWidth|mask|maskContentUnits|maskUnits|mathematical|mode|numOctaves|offset|opacity|operator|order|orient|orientation|origin|overflow|overlinePosition|overlineThickness|panose1|paintOrder|pathLength|patternContentUnits|patternTransform|patternUnits|pointerEvents|points|pointsAtX|pointsAtY|pointsAtZ|preserveAlpha|preserveAspectRatio|primitiveUnits|r|radius|refX|refY|renderingIntent|repeatCount|repeatDur|requiredExtensions|requiredFeatures|restart|result|rotate|rx|ry|scale|seed|shapeRendering|slope|spacing|specularConstant|specularExponent|speed|spreadMethod|startOffset|stdDeviation|stemh|stemv|stitchTiles|stopColor|stopOpacity|strikethroughPosition|strikethroughThickness|string|stroke|strokeDasharray|strokeDashoffset|strokeLinecap|strokeLinejoin|strokeMiterlimit|strokeOpacity|strokeWidth|surfaceScale|systemLanguage|tableValues|targetX|targetY|textAnchor|textDecoration|textRendering|textLength|to|transform|u1|u2|underlinePosition|underlineThickness|unicode|unicodeBidi|unicodeRange|unitsPerEm|vAlphabetic|vHanging|vIdeographic|vMathematical|values|vectorEffect|version|vertAdvY|vertOriginX|vertOriginY|viewBox|viewTarget|visibility|widths|wordSpacing|writingMode|x|xHeight|x1|x2|xChannelSelector|xlinkActuate|xlinkArcrole|xlinkHref|xlinkRole|xlinkShow|xlinkTitle|xlinkType|xmlBase|xmlns|xmlnsXlink|xmlLang|xmlSpace|y|y1|y2|yChannelSelector|z|zoomAndPan|for|class|autofocus)|(([Dd][Aa][Tt][Aa]|[Aa][Rr][Ii][Aa]|x)-.*))$/, Ot = /* @__PURE__ */ xr(
  function(e) {
    return Ar.test(e) || e.charCodeAt(0) === 111 && e.charCodeAt(1) === 110 && e.charCodeAt(2) < 91;
  }
), mt = Ge.exports, kr = {
  childContextTypes: !0,
  contextType: !0,
  contextTypes: !0,
  defaultProps: !0,
  displayName: !0,
  getDefaultProps: !0,
  getDerivedStateFromError: !0,
  getDerivedStateFromProps: !0,
  mixins: !0,
  propTypes: !0,
  type: !0
}, Tr = {
  name: !0,
  length: !0,
  prototype: !0,
  caller: !0,
  callee: !0,
  arguments: !0,
  arity: !0
}, Pr = {
  $$typeof: !0,
  render: !0,
  defaultProps: !0,
  displayName: !0,
  propTypes: !0
}, Ht = {
  $$typeof: !0,
  compare: !0,
  defaultProps: !0,
  displayName: !0,
  propTypes: !0,
  type: !0
}, vt = {};
vt[mt.ForwardRef] = Pr;
vt[mt.Memo] = Ht;
function It(e) {
  return mt.isMemo(e) ? Ht : vt[e.$$typeof] || kr;
}
var Or = Object.defineProperty, Ir = Object.getOwnPropertyNames, Nt = Object.getOwnPropertySymbols, Nr = Object.getOwnPropertyDescriptor, Dr = Object.getPrototypeOf, Dt = Object.prototype;
function Gt(e, n, r) {
  if (typeof n != "string") {
    if (Dt) {
      var o = Dr(n);
      o && o !== Dt && Gt(e, o, r);
    }
    var i = Ir(n);
    Nt && (i = i.concat(Nt(n)));
    for (var s = It(e), c = It(n), p = 0; p < i.length; ++p) {
      var h = i[p];
      if (!Tr[h] && !(r && r[h]) && !(c && c[h]) && !(s && s[h])) {
        var b = Nr(n, h);
        try {
          Or(e, h, b);
        } catch {
        }
      }
    }
  }
  return e;
}
var Mr = Gt;
function ke() {
  return (ke = Object.assign || function(e) {
    for (var n = 1; n < arguments.length; n++) {
      var r = arguments[n];
      for (var o in r)
        Object.prototype.hasOwnProperty.call(r, o) && (e[o] = r[o]);
    }
    return e;
  }).apply(this, arguments);
}
var Mt = function(e, n) {
  for (var r = [e[0]], o = 0, i = n.length; o < i; o += 1)
    r.push(n[o], e[o + 1]);
  return r;
}, lt = function(e) {
  return e !== null && typeof e == "object" && (e.toString ? e.toString() : Object.prototype.toString.call(e)) === "[object Object]" && !Ge.exports.typeOf(e);
}, Qe = Object.freeze([]), Te = Object.freeze({});
function Be(e) {
  return typeof e == "function";
}
function ut(e) {
  return process.env.NODE_ENV !== "production" && typeof e == "string" && e || e.displayName || e.name || "Component";
}
function gt(e) {
  return e && typeof e.styledComponentId == "string";
}
var We = typeof process < "u" && (process.env.REACT_APP_SC_ATTR || process.env.SC_ATTR) || "data-styled", yt = typeof window < "u" && "HTMLElement" in window, jr = Boolean(typeof SC_DISABLE_SPEEDY == "boolean" ? SC_DISABLE_SPEEDY : typeof process < "u" && process.env.REACT_APP_SC_DISABLE_SPEEDY !== void 0 && process.env.REACT_APP_SC_DISABLE_SPEEDY !== "" ? process.env.REACT_APP_SC_DISABLE_SPEEDY !== "false" && process.env.REACT_APP_SC_DISABLE_SPEEDY : typeof process < "u" && process.env.SC_DISABLE_SPEEDY !== void 0 && process.env.SC_DISABLE_SPEEDY !== "" ? process.env.SC_DISABLE_SPEEDY !== "false" && process.env.SC_DISABLE_SPEEDY : process.env.NODE_ENV !== "production"), Wr = process.env.NODE_ENV !== "production" ? { 1: `Cannot create styled-component for component: %s.

`, 2: `Can't collect styles once you've consumed a \`ServerStyleSheet\`'s styles! \`ServerStyleSheet\` is a one off instance for each server-side render cycle.

- Are you trying to reuse it across renders?
- Are you accidentally calling collectStyles twice?

`, 3: `Streaming SSR is only supported in a Node.js environment; Please do not try to call this method in the browser.

`, 4: `The \`StyleSheetManager\` expects a valid target or sheet prop!

- Does this error occur on the client and is your target falsy?
- Does this error occur on the server and is the sheet falsy?

`, 5: `The clone method cannot be used on the client!

- Are you running in a client-like environment on the server?
- Are you trying to run SSR on the client?

`, 6: `Trying to insert a new style tag, but the given Node is unmounted!

- Are you using a custom target that isn't mounted?
- Does your document not have a valid head element?
- Have you accidentally removed a style tag manually?

`, 7: 'ThemeProvider: Please return an object from your "theme" prop function, e.g.\n\n```js\ntheme={() => ({})}\n```\n\n', 8: `ThemeProvider: Please make your "theme" prop an object.

`, 9: "Missing document `<head>`\n\n", 10: `Cannot find a StyleSheet instance. Usually this happens if there are multiple copies of styled-components loaded at once. Check out this issue for how to troubleshoot and fix the common cases where this situation can happen: https://github.com/styled-components/styled-components/issues/1941#issuecomment-417862021

`, 11: `_This error was replaced with a dev-time warning, it will be deleted for v4 final._ [createGlobalStyle] received children which will not be rendered. Please use the component without passing children elements.

`, 12: "It seems you are interpolating a keyframe declaration (%s) into an untagged string. This was supported in styled-components v3, but is not longer supported in v4 as keyframes are now injected on-demand. Please wrap your string in the css\\`\\` helper which ensures the styles are injected correctly. See https://www.styled-components.com/docs/api#css\n\n", 13: `%s is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.

`, 14: `ThemeProvider: "theme" prop is required.

`, 15: "A stylis plugin has been supplied that is not named. We need a name for each plugin to be able to prevent styling collisions between different stylis configurations within the same app. Before you pass your plugin to `<StyleSheetManager stylisPlugins={[]}>`, please make sure each plugin is uniquely-named, e.g.\n\n```js\nObject.defineProperty(importedPlugin, 'name', { value: 'some-unique-name' });\n```\n\n", 16: `Reached the limit of how many styled components may be created at group %s.
You may only create up to 1,073,741,824 components. If you're creating components dynamically,
as for instance in your render method then you may be running into this limitation.

`, 17: `CSSStyleSheet could not be found on HTMLStyleElement.
Has styled-components' style tag been unmounted or altered by another script?
` } : {};
function $r() {
  for (var e = arguments.length <= 0 ? void 0 : arguments[0], n = [], r = 1, o = arguments.length; r < o; r += 1)
    n.push(r < 0 || arguments.length <= r ? void 0 : arguments[r]);
  return n.forEach(function(i) {
    e = e.replace(/%[a-z]/, i);
  }), e;
}
function Fe(e) {
  for (var n = arguments.length, r = new Array(n > 1 ? n - 1 : 0), o = 1; o < n; o++)
    r[o - 1] = arguments[o];
  throw process.env.NODE_ENV === "production" ? new Error("An error occurred. See https://git.io/JUIaE#" + e + " for more information." + (r.length > 0 ? " Args: " + r.join(", ") : "")) : new Error($r.apply(void 0, [Wr[e]].concat(r)).trim());
}
var Lr = function() {
  function e(r) {
    this.groupSizes = new Uint32Array(512), this.length = 512, this.tag = r;
  }
  var n = e.prototype;
  return n.indexOfGroup = function(r) {
    for (var o = 0, i = 0; i < r; i++)
      o += this.groupSizes[i];
    return o;
  }, n.insertRules = function(r, o) {
    if (r >= this.groupSizes.length) {
      for (var i = this.groupSizes, s = i.length, c = s; r >= c; )
        (c <<= 1) < 0 && Fe(16, "" + r);
      this.groupSizes = new Uint32Array(c), this.groupSizes.set(i), this.length = c;
      for (var p = s; p < c; p++)
        this.groupSizes[p] = 0;
    }
    for (var h = this.indexOfGroup(r + 1), b = 0, g = o.length; b < g; b++)
      this.tag.insertRule(h, o[b]) && (this.groupSizes[r]++, h++);
  }, n.clearGroup = function(r) {
    if (r < this.length) {
      var o = this.groupSizes[r], i = this.indexOfGroup(r), s = i + o;
      this.groupSizes[r] = 0;
      for (var c = i; c < s; c++)
        this.tag.deleteRule(i);
    }
  }, n.getGroup = function(r) {
    var o = "";
    if (r >= this.length || this.groupSizes[r] === 0)
      return o;
    for (var i = this.groupSizes[r], s = this.indexOfGroup(r), c = s + i, p = s; p < c; p++)
      o += this.tag.getRule(p) + `/*!sc*/
`;
    return o;
  }, e;
}(), Je = /* @__PURE__ */ new Map(), et = /* @__PURE__ */ new Map(), Ve = 1, qe = function(e) {
  if (Je.has(e))
    return Je.get(e);
  for (; et.has(Ve); )
    Ve++;
  var n = Ve++;
  return process.env.NODE_ENV !== "production" && ((0 | n) < 0 || n > 1 << 30) && Fe(16, "" + n), Je.set(e, n), et.set(n, e), n;
}, Fr = function(e) {
  return et.get(e);
}, zr = function(e, n) {
  n >= Ve && (Ve = n + 1), Je.set(e, n), et.set(n, e);
}, Yr = "style[" + We + '][data-styled-version="5.3.6"]', Vr = new RegExp("^" + We + '\\.g(\\d+)\\[id="([\\w\\d-]+)"\\].*?"([^"]*)'), Br = function(e, n, r) {
  for (var o, i = r.split(","), s = 0, c = i.length; s < c; s++)
    (o = i[s]) && e.registerName(n, o);
}, Ur = function(e, n) {
  for (var r = (n.textContent || "").split(`/*!sc*/
`), o = [], i = 0, s = r.length; i < s; i++) {
    var c = r[i].trim();
    if (c) {
      var p = c.match(Vr);
      if (p) {
        var h = 0 | parseInt(p[1], 10), b = p[2];
        h !== 0 && (zr(b, h), Br(e, b, p[3]), e.getTag().insertRules(h, o)), o.length = 0;
      } else
        o.push(c);
    }
  }
}, Hr = function() {
  return typeof __webpack_nonce__ < "u" ? __webpack_nonce__ : null;
}, qt = function(e) {
  var n = document.head, r = e || n, o = document.createElement("style"), i = function(p) {
    for (var h = p.childNodes, b = h.length; b >= 0; b--) {
      var g = h[b];
      if (g && g.nodeType === 1 && g.hasAttribute(We))
        return g;
    }
  }(r), s = i !== void 0 ? i.nextSibling : null;
  o.setAttribute(We, "active"), o.setAttribute("data-styled-version", "5.3.6");
  var c = Hr();
  return c && o.setAttribute("nonce", c), r.insertBefore(o, s), o;
}, Gr = function() {
  function e(r) {
    var o = this.element = qt(r);
    o.appendChild(document.createTextNode("")), this.sheet = function(i) {
      if (i.sheet)
        return i.sheet;
      for (var s = document.styleSheets, c = 0, p = s.length; c < p; c++) {
        var h = s[c];
        if (h.ownerNode === i)
          return h;
      }
      Fe(17);
    }(o), this.length = 0;
  }
  var n = e.prototype;
  return n.insertRule = function(r, o) {
    try {
      return this.sheet.insertRule(o, r), this.length++, !0;
    } catch {
      return !1;
    }
  }, n.deleteRule = function(r) {
    this.sheet.deleteRule(r), this.length--;
  }, n.getRule = function(r) {
    var o = this.sheet.cssRules[r];
    return o !== void 0 && typeof o.cssText == "string" ? o.cssText : "";
  }, e;
}(), qr = function() {
  function e(r) {
    var o = this.element = qt(r);
    this.nodes = o.childNodes, this.length = 0;
  }
  var n = e.prototype;
  return n.insertRule = function(r, o) {
    if (r <= this.length && r >= 0) {
      var i = document.createTextNode(o), s = this.nodes[r];
      return this.element.insertBefore(i, s || null), this.length++, !0;
    }
    return !1;
  }, n.deleteRule = function(r) {
    this.element.removeChild(this.nodes[r]), this.length--;
  }, n.getRule = function(r) {
    return r < this.length ? this.nodes[r].textContent : "";
  }, e;
}(), Xr = function() {
  function e(r) {
    this.rules = [], this.length = 0;
  }
  var n = e.prototype;
  return n.insertRule = function(r, o) {
    return r <= this.length && (this.rules.splice(r, 0, o), this.length++, !0);
  }, n.deleteRule = function(r) {
    this.rules.splice(r, 1), this.length--;
  }, n.getRule = function(r) {
    return r < this.length ? this.rules[r] : "";
  }, e;
}(), jt = yt, Kr = { isServer: !yt, useCSSOMInjection: !jr }, Xt = function() {
  function e(r, o, i) {
    r === void 0 && (r = Te), o === void 0 && (o = {}), this.options = ke({}, Kr, {}, r), this.gs = o, this.names = new Map(i), this.server = !!r.isServer, !this.server && yt && jt && (jt = !1, function(s) {
      for (var c = document.querySelectorAll(Yr), p = 0, h = c.length; p < h; p++) {
        var b = c[p];
        b && b.getAttribute(We) !== "active" && (Ur(s, b), b.parentNode && b.parentNode.removeChild(b));
      }
    }(this));
  }
  e.registerId = function(r) {
    return qe(r);
  };
  var n = e.prototype;
  return n.reconstructWithOptions = function(r, o) {
    return o === void 0 && (o = !0), new e(ke({}, this.options, {}, r), this.gs, o && this.names || void 0);
  }, n.allocateGSInstance = function(r) {
    return this.gs[r] = (this.gs[r] || 0) + 1;
  }, n.getTag = function() {
    return this.tag || (this.tag = (i = (o = this.options).isServer, s = o.useCSSOMInjection, c = o.target, r = i ? new Xr(c) : s ? new Gr(c) : new qr(c), new Lr(r)));
    var r, o, i, s, c;
  }, n.hasNameForId = function(r, o) {
    return this.names.has(r) && this.names.get(r).has(o);
  }, n.registerName = function(r, o) {
    if (qe(r), this.names.has(r))
      this.names.get(r).add(o);
    else {
      var i = /* @__PURE__ */ new Set();
      i.add(o), this.names.set(r, i);
    }
  }, n.insertRules = function(r, o, i) {
    this.registerName(r, o), this.getTag().insertRules(qe(r), i);
  }, n.clearNames = function(r) {
    this.names.has(r) && this.names.get(r).clear();
  }, n.clearRules = function(r) {
    this.getTag().clearGroup(qe(r)), this.clearNames(r);
  }, n.clearTag = function() {
    this.tag = void 0;
  }, n.toString = function() {
    return function(r) {
      for (var o = r.getTag(), i = o.length, s = "", c = 0; c < i; c++) {
        var p = Fr(c);
        if (p !== void 0) {
          var h = r.names.get(p), b = o.getGroup(c);
          if (h && b && h.size) {
            var g = We + ".g" + c + '[id="' + p + '"]', E = "";
            h !== void 0 && h.forEach(function(L) {
              L.length > 0 && (E += L + ",");
            }), s += "" + b + g + '{content:"' + E + `"}/*!sc*/
`;
          }
        }
      }
      return s;
    }(this);
  }, e;
}(), Zr = /(a)(d)/gi, Wt = function(e) {
  return String.fromCharCode(e + (e > 25 ? 39 : 97));
};
function dt(e) {
  var n, r = "";
  for (n = Math.abs(e); n > 52; n = n / 52 | 0)
    r = Wt(n % 52) + r;
  return (Wt(n % 52) + r).replace(Zr, "$1-$2");
}
var Oe = function(e, n) {
  for (var r = n.length; r; )
    e = 33 * e ^ n.charCodeAt(--r);
  return e;
}, Kt = function(e) {
  return Oe(5381, e);
};
function Jr(e) {
  for (var n = 0; n < e.length; n += 1) {
    var r = e[n];
    if (Be(r) && !gt(r))
      return !1;
  }
  return !0;
}
var Qr = Kt("5.3.6"), en = function() {
  function e(n, r, o) {
    this.rules = n, this.staticRulesId = "", this.isStatic = process.env.NODE_ENV === "production" && (o === void 0 || o.isStatic) && Jr(n), this.componentId = r, this.baseHash = Oe(Qr, r), this.baseStyle = o, Xt.registerId(r);
  }
  return e.prototype.generateAndInjectStyles = function(n, r, o) {
    var i = this.componentId, s = [];
    if (this.baseStyle && s.push(this.baseStyle.generateAndInjectStyles(n, r, o)), this.isStatic && !o.hash)
      if (this.staticRulesId && r.hasNameForId(i, this.staticRulesId))
        s.push(this.staticRulesId);
      else {
        var c = $e(this.rules, n, r, o).join(""), p = dt(Oe(this.baseHash, c) >>> 0);
        if (!r.hasNameForId(i, p)) {
          var h = o(c, "." + p, void 0, i);
          r.insertRules(i, p, h);
        }
        s.push(p), this.staticRulesId = p;
      }
    else {
      for (var b = this.rules.length, g = Oe(this.baseHash, o.hash), E = "", L = 0; L < b; L++) {
        var F = this.rules[L];
        if (typeof F == "string")
          E += F, process.env.NODE_ENV !== "production" && (g = Oe(g, F + L));
        else if (F) {
          var x = $e(F, n, r, o), I = Array.isArray(x) ? x.join("") : x;
          g = Oe(g, I + L), E += I;
        }
      }
      if (E) {
        var k = dt(g >>> 0);
        if (!r.hasNameForId(i, k)) {
          var N = o(E, "." + k, void 0, i);
          r.insertRules(i, k, N);
        }
        s.push(k);
      }
    }
    return s.join(" ");
  }, e;
}(), tn = /^\s*\/\/.*$/gm, rn = [":", "[", ".", "#"];
function nn(e) {
  var n, r, o, i, s = e === void 0 ? Te : e, c = s.options, p = c === void 0 ? Te : c, h = s.plugins, b = h === void 0 ? Qe : h, g = new _r(p), E = [], L = function(I) {
    function k(N) {
      if (N)
        try {
          I(N + "}");
        } catch {
        }
    }
    return function(N, S, T, C, d, oe, pe, se, le, he) {
      switch (N) {
        case 1:
          if (le === 0 && S.charCodeAt(0) === 64)
            return I(S + ";"), "";
          break;
        case 2:
          if (se === 0)
            return S + "/*|*/";
          break;
        case 3:
          switch (se) {
            case 102:
            case 112:
              return I(T[0] + S), "";
            default:
              return S + (he === 0 ? "/*|*/" : "");
          }
        case -2:
          S.split("/*|*/}").forEach(k);
      }
    };
  }(function(I) {
    E.push(I);
  }), F = function(I, k, N) {
    return k === 0 && rn.indexOf(N[r.length]) !== -1 || N.match(i) ? I : "." + n;
  };
  function x(I, k, N, S) {
    S === void 0 && (S = "&");
    var T = I.replace(tn, ""), C = k && N ? N + " " + k + " { " + T + " }" : T;
    return n = S, r = k, o = new RegExp("\\" + r + "\\b", "g"), i = new RegExp("(\\" + r + "\\b){2,}"), g(N || !k ? "" : k, C);
  }
  return g.use([].concat(b, [function(I, k, N) {
    I === 2 && N.length && N[0].lastIndexOf(r) > 0 && (N[0] = N[0].replace(o, F));
  }, L, function(I) {
    if (I === -2) {
      var k = E;
      return E = [], k;
    }
  }])), x.hash = b.length ? b.reduce(function(I, k) {
    return k.name || Fe(15), Oe(I, k.name);
  }, 5381).toString() : "", x;
}
var Zt = Le.createContext();
Zt.Consumer;
var Jt = Le.createContext(), on = (Jt.Consumer, new Xt()), ft = nn();
function an() {
  return _e(Zt) || on;
}
function sn() {
  return _e(Jt) || ft;
}
var cn = function() {
  function e(n, r) {
    var o = this;
    this.inject = function(i, s) {
      s === void 0 && (s = ft);
      var c = o.name + s.hash;
      i.hasNameForId(o.id, c) || i.insertRules(o.id, c, s(o.rules, c, "@keyframes"));
    }, this.toString = function() {
      return Fe(12, String(o.name));
    }, this.name = n, this.id = "sc-keyframes-" + n, this.rules = r;
  }
  return e.prototype.getName = function(n) {
    return n === void 0 && (n = ft), this.name + n.hash;
  }, e;
}(), ln = /([A-Z])/, un = /([A-Z])/g, dn = /^ms-/, fn = function(e) {
  return "-" + e.toLowerCase();
};
function $t(e) {
  return ln.test(e) ? e.replace(un, fn).replace(dn, "-ms-") : e;
}
var Lt = function(e) {
  return e == null || e === !1 || e === "";
};
function $e(e, n, r, o) {
  if (Array.isArray(e)) {
    for (var i, s = [], c = 0, p = e.length; c < p; c += 1)
      (i = $e(e[c], n, r, o)) !== "" && (Array.isArray(i) ? s.push.apply(s, i) : s.push(i));
    return s;
  }
  if (Lt(e))
    return "";
  if (gt(e))
    return "." + e.styledComponentId;
  if (Be(e)) {
    if (typeof (b = e) != "function" || b.prototype && b.prototype.isReactComponent || !n)
      return e;
    var h = e(n);
    return process.env.NODE_ENV !== "production" && Ge.exports.isElement(h) && console.warn(ut(e) + " is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details."), $e(h, n, r, o);
  }
  var b;
  return e instanceof cn ? r ? (e.inject(r, o), e.getName(o)) : e : lt(e) ? function g(E, L) {
    var F, x, I = [];
    for (var k in E)
      E.hasOwnProperty(k) && !Lt(E[k]) && (Array.isArray(E[k]) && E[k].isCss || Be(E[k]) ? I.push($t(k) + ":", E[k], ";") : lt(E[k]) ? I.push.apply(I, g(E[k], k)) : I.push($t(k) + ": " + (F = k, (x = E[k]) == null || typeof x == "boolean" || x === "" ? "" : typeof x != "number" || x === 0 || F in Rr ? String(x).trim() : x + "px") + ";"));
    return L ? [L + " {"].concat(I, ["}"]) : I;
  }(e) : e.toString();
}
var Ft = function(e) {
  return Array.isArray(e) && (e.isCss = !0), e;
};
function pn(e) {
  for (var n = arguments.length, r = new Array(n > 1 ? n - 1 : 0), o = 1; o < n; o++)
    r[o - 1] = arguments[o];
  return Be(e) || lt(e) ? Ft($e(Mt(Qe, [e].concat(r)))) : r.length === 0 && e.length === 1 && typeof e[0] == "string" ? e : Ft($e(Mt(e, r)));
}
var zt = /invalid hook call/i, Xe = /* @__PURE__ */ new Set(), hn = function(e, n) {
  if (process.env.NODE_ENV !== "production") {
    var r = "The component " + e + (n ? ' with the id of "' + n + '"' : "") + ` has been created dynamically.
You may see this warning because you've called styled inside another component.
To resolve this only create new StyledComponents outside of any render method and function component.`, o = console.error;
    try {
      var i = !0;
      console.error = function(s) {
        if (zt.test(s))
          i = !1, Xe.delete(r);
        else {
          for (var c = arguments.length, p = new Array(c > 1 ? c - 1 : 0), h = 1; h < c; h++)
            p[h - 1] = arguments[h];
          o.apply(void 0, [s].concat(p));
        }
      }, yr(), i && !Xe.has(r) && (console.warn(r), Xe.add(r));
    } catch (s) {
      zt.test(s.message) && Xe.delete(r);
    } finally {
      console.error = o;
    }
  }
}, mn = function(e, n, r) {
  return r === void 0 && (r = Te), e.theme !== r.theme && e.theme || n || r.theme;
}, vn = /[!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~-]+/g, gn = /(^-|-$)/g;
function it(e) {
  return e.replace(vn, "-").replace(gn, "");
}
var yn = function(e) {
  return dt(Kt(e) >>> 0);
};
function Ke(e) {
  return typeof e == "string" && (process.env.NODE_ENV === "production" || e.charAt(0) === e.charAt(0).toLowerCase());
}
var pt = function(e) {
  return typeof e == "function" || typeof e == "object" && e !== null && !Array.isArray(e);
}, wn = function(e) {
  return e !== "__proto__" && e !== "constructor" && e !== "prototype";
};
function bn(e, n, r) {
  var o = e[r];
  pt(n) && pt(o) ? Qt(o, n) : e[r] = n;
}
function Qt(e) {
  for (var n = arguments.length, r = new Array(n > 1 ? n - 1 : 0), o = 1; o < n; o++)
    r[o - 1] = arguments[o];
  for (var i = 0, s = r; i < s.length; i++) {
    var c = s[i];
    if (pt(c))
      for (var p in c)
        wn(p) && bn(e, c[p], p);
  }
  return e;
}
var er = Le.createContext();
er.Consumer;
var st = {};
function tr(e, n, r) {
  var o = gt(e), i = !Ke(e), s = n.attrs, c = s === void 0 ? Qe : s, p = n.componentId, h = p === void 0 ? function(S, T) {
    var C = typeof S != "string" ? "sc" : it(S);
    st[C] = (st[C] || 0) + 1;
    var d = C + "-" + yn("5.3.6" + C + st[C]);
    return T ? T + "-" + d : d;
  }(n.displayName, n.parentComponentId) : p, b = n.displayName, g = b === void 0 ? function(S) {
    return Ke(S) ? "styled." + S : "Styled(" + ut(S) + ")";
  }(e) : b, E = n.displayName && n.componentId ? it(n.displayName) + "-" + n.componentId : n.componentId || h, L = o && e.attrs ? Array.prototype.concat(e.attrs, c).filter(Boolean) : c, F = n.shouldForwardProp;
  o && e.shouldForwardProp && (F = n.shouldForwardProp ? function(S, T, C) {
    return e.shouldForwardProp(S, T, C) && n.shouldForwardProp(S, T, C);
  } : e.shouldForwardProp);
  var x, I = new en(r, E, o ? e.componentStyle : void 0), k = I.isStatic && c.length === 0, N = function(S, T) {
    return function(C, d, oe, pe) {
      var se = C.attrs, le = C.componentStyle, he = C.defaultProps, ye = C.foldedComponentIds, ae = C.shouldForwardProp, H = C.styledComponentId, ee = C.target;
      process.env.NODE_ENV !== "production" && kt(H);
      var K = function(A, a, D) {
        A === void 0 && (A = Te);
        var f = ke({}, a, { theme: A }), B = {};
        return D.forEach(function(O) {
          var j, l, q, Z = O;
          for (j in Be(Z) && (Z = Z(f)), Z)
            f[j] = B[j] = j === "className" ? (l = B[j], q = Z[j], l && q ? l + " " + q : l || q) : Z[j];
        }), [f, B];
      }(mn(d, _e(er), he) || Te, d, se), we = K[0], ce = K[1], ie = function(A, a, D, f) {
        var B = an(), O = sn(), j = a ? A.generateAndInjectStyles(Te, B, O) : A.generateAndInjectStyles(D, B, O);
        return process.env.NODE_ENV !== "production" && kt(j), process.env.NODE_ENV !== "production" && !a && f && f(j), j;
      }(le, pe, we, process.env.NODE_ENV !== "production" ? C.warnTooManyClasses : void 0), be = oe, me = ce.$as || d.$as || ce.as || d.as || ee, Se = Ke(me), w = ce !== d ? ke({}, d, {}, ce) : d, v = {};
      for (var y in w)
        y[0] !== "$" && y !== "as" && (y === "forwardedAs" ? v.as = w[y] : (ae ? ae(y, Ot, me) : !Se || Ot(y)) && (v[y] = w[y]));
      return d.style && ce.style !== d.style && (v.style = ke({}, d.style, {}, ce.style)), v.className = Array.prototype.concat(ye, H, ie !== H ? ie : null, d.className, ce.className).filter(Boolean).join(" "), v.ref = be, wr(me, v);
    }(x, S, T, k);
  };
  return N.displayName = g, (x = Le.forwardRef(N)).attrs = L, x.componentStyle = I, x.displayName = g, x.shouldForwardProp = F, x.foldedComponentIds = o ? Array.prototype.concat(e.foldedComponentIds, e.styledComponentId) : Qe, x.styledComponentId = E, x.target = o ? e.target : e, x.withComponent = function(S) {
    var T = n.componentId, C = function(oe, pe) {
      if (oe == null)
        return {};
      var se, le, he = {}, ye = Object.keys(oe);
      for (le = 0; le < ye.length; le++)
        se = ye[le], pe.indexOf(se) >= 0 || (he[se] = oe[se]);
      return he;
    }(n, ["componentId"]), d = T && T + "-" + (Ke(S) ? S : it(ut(S)));
    return tr(S, ke({}, C, { attrs: L, componentId: d }), r);
  }, Object.defineProperty(x, "defaultProps", { get: function() {
    return this._foldedDefaultProps;
  }, set: function(S) {
    this._foldedDefaultProps = o ? Qt({}, e.defaultProps, S) : S;
  } }), process.env.NODE_ENV !== "production" && (hn(g, E), x.warnTooManyClasses = function(S, T) {
    var C = {}, d = !1;
    return function(oe) {
      if (!d && (C[oe] = !0, Object.keys(C).length >= 200)) {
        var pe = T ? ' with the id of "' + T + '"' : "";
        console.warn("Over 200 classes were generated for component " + S + pe + `.
Consider using the attrs method, together with a style object for frequently changed styles.
Example:
  const Component = styled.div.attrs(props => ({
    style: {
      background: props.background,
    },
  }))\`width: 100%;\`

  <Component />`), d = !0, C = {};
      }
    };
  }(g, E)), x.toString = function() {
    return "." + x.styledComponentId;
  }, i && Mr(x, e, { attrs: !0, componentStyle: !0, displayName: !0, foldedComponentIds: !0, shouldForwardProp: !0, styledComponentId: !0, target: !0, withComponent: !0 }), x;
}
var ht = function(e) {
  return function n(r, o, i) {
    if (i === void 0 && (i = Te), !Ge.exports.isValidElementType(o))
      return Fe(1, String(o));
    var s = function() {
      return r(o, i, pn.apply(void 0, arguments));
    };
    return s.withConfig = function(c) {
      return n(r, o, ke({}, i, {}, c));
    }, s.attrs = function(c) {
      return n(r, o, ke({}, i, { attrs: Array.prototype.concat(i.attrs, c).filter(Boolean) }));
    }, s;
  }(tr, e);
};
["a", "abbr", "address", "area", "article", "aside", "audio", "b", "base", "bdi", "bdo", "big", "blockquote", "body", "br", "button", "canvas", "caption", "cite", "code", "col", "colgroup", "data", "datalist", "dd", "del", "details", "dfn", "dialog", "div", "dl", "dt", "em", "embed", "fieldset", "figcaption", "figure", "footer", "form", "h1", "h2", "h3", "h4", "h5", "h6", "head", "header", "hgroup", "hr", "html", "i", "iframe", "img", "input", "ins", "kbd", "keygen", "label", "legend", "li", "link", "main", "map", "mark", "marquee", "menu", "menuitem", "meta", "meter", "nav", "noscript", "object", "ol", "optgroup", "option", "output", "p", "param", "picture", "pre", "progress", "q", "rp", "rt", "ruby", "s", "samp", "script", "section", "select", "small", "source", "span", "strong", "style", "sub", "summary", "sup", "table", "tbody", "td", "textarea", "tfoot", "th", "thead", "time", "title", "tr", "track", "u", "ul", "var", "video", "wbr", "circle", "clipPath", "defs", "ellipse", "foreignObject", "g", "image", "line", "linearGradient", "marker", "mask", "path", "pattern", "polygon", "polyline", "radialGradient", "rect", "stop", "svg", "text", "textPath", "tspan"].forEach(function(e) {
  ht[e] = ht(e);
});
process.env.NODE_ENV !== "production" && typeof navigator < "u" && navigator.product === "ReactNative" && console.warn(`It looks like you've imported 'styled-components' on React Native.
Perhaps you're looking to import 'styled-components/native'?
Read more about this at https://www.styled-components.com/docs/basics#react-native`), process.env.NODE_ENV !== "production" && process.env.NODE_ENV !== "test" && typeof window < "u" && (window["__styled-components-init__"] = window["__styled-components-init__"] || 0, window["__styled-components-init__"] === 1 && console.warn(`It looks like there are several instances of 'styled-components' initialized in this application. This may cause dynamic styles to not render properly, errors during the rehydration process, a missing theme prop, and makes your application bigger without good reason.

See https://s-c.sh/2BAXzed for more info.`), window["__styled-components-init__"] += 1);
const ge = ht;
var tt = { exports: {} }, ze = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Yt;
function Sn() {
  if (Yt)
    return ze;
  Yt = 1;
  var e = Le, n = Symbol.for("react.element"), r = Symbol.for("react.fragment"), o = Object.prototype.hasOwnProperty, i = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, s = { key: !0, ref: !0, __self: !0, __source: !0 };
  function c(p, h, b) {
    var g, E = {}, L = null, F = null;
    b !== void 0 && (L = "" + b), h.key !== void 0 && (L = "" + h.key), h.ref !== void 0 && (F = h.ref);
    for (g in h)
      o.call(h, g) && !s.hasOwnProperty(g) && (E[g] = h[g]);
    if (p && p.defaultProps)
      for (g in h = p.defaultProps, h)
        E[g] === void 0 && (E[g] = h[g]);
    return { $$typeof: n, type: p, key: L, ref: F, props: E, _owner: i.current };
  }
  return ze.Fragment = r, ze.jsx = c, ze.jsxs = c, ze;
}
var Ye = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Vt;
function En() {
  return Vt || (Vt = 1, process.env.NODE_ENV !== "production" && function() {
    var e = Le, n = Symbol.for("react.element"), r = Symbol.for("react.portal"), o = Symbol.for("react.fragment"), i = Symbol.for("react.strict_mode"), s = Symbol.for("react.profiler"), c = Symbol.for("react.provider"), p = Symbol.for("react.context"), h = Symbol.for("react.forward_ref"), b = Symbol.for("react.suspense"), g = Symbol.for("react.suspense_list"), E = Symbol.for("react.memo"), L = Symbol.for("react.lazy"), F = Symbol.for("react.offscreen"), x = Symbol.iterator, I = "@@iterator";
    function k(t) {
      if (t === null || typeof t != "object")
        return null;
      var u = x && t[x] || t[I];
      return typeof u == "function" ? u : null;
    }
    var N = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function S(t) {
      {
        for (var u = arguments.length, m = new Array(u > 1 ? u - 1 : 0), R = 1; R < u; R++)
          m[R - 1] = arguments[R];
        T("error", t, m);
      }
    }
    function T(t, u, m) {
      {
        var R = N.ReactDebugCurrentFrame, z = R.getStackAddendum();
        z !== "" && (u += "%s", m = m.concat([z]));
        var U = m.map(function(W) {
          return String(W);
        });
        U.unshift("Warning: " + u), Function.prototype.apply.call(console[t], console, U);
      }
    }
    var C = !1, d = !1, oe = !1, pe = !1, se = !1, le;
    le = Symbol.for("react.module.reference");
    function he(t) {
      return !!(typeof t == "string" || typeof t == "function" || t === o || t === s || se || t === i || t === b || t === g || pe || t === F || C || d || oe || typeof t == "object" && t !== null && (t.$$typeof === L || t.$$typeof === E || t.$$typeof === c || t.$$typeof === p || t.$$typeof === h || t.$$typeof === le || t.getModuleId !== void 0));
    }
    function ye(t, u, m) {
      var R = t.displayName;
      if (R)
        return R;
      var z = u.displayName || u.name || "";
      return z !== "" ? m + "(" + z + ")" : m;
    }
    function ae(t) {
      return t.displayName || "Context";
    }
    function H(t) {
      if (t == null)
        return null;
      if (typeof t.tag == "number" && S("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof t == "function")
        return t.displayName || t.name || null;
      if (typeof t == "string")
        return t;
      switch (t) {
        case o:
          return "Fragment";
        case r:
          return "Portal";
        case s:
          return "Profiler";
        case i:
          return "StrictMode";
        case b:
          return "Suspense";
        case g:
          return "SuspenseList";
      }
      if (typeof t == "object")
        switch (t.$$typeof) {
          case p:
            var u = t;
            return ae(u) + ".Consumer";
          case c:
            var m = t;
            return ae(m._context) + ".Provider";
          case h:
            return ye(t, t.render, "ForwardRef");
          case E:
            var R = t.displayName || null;
            return R !== null ? R : H(t.type) || "Memo";
          case L: {
            var z = t, U = z._payload, W = z._init;
            try {
              return H(W(U));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var ee = Object.assign, K = 0, we, ce, ie, be, me, Se, w;
    function v() {
    }
    v.__reactDisabledLog = !0;
    function y() {
      {
        if (K === 0) {
          we = console.log, ce = console.info, ie = console.warn, be = console.error, me = console.group, Se = console.groupCollapsed, w = console.groupEnd;
          var t = {
            configurable: !0,
            enumerable: !0,
            value: v,
            writable: !0
          };
          Object.defineProperties(console, {
            info: t,
            log: t,
            warn: t,
            error: t,
            group: t,
            groupCollapsed: t,
            groupEnd: t
          });
        }
        K++;
      }
    }
    function A() {
      {
        if (K--, K === 0) {
          var t = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: ee({}, t, {
              value: we
            }),
            info: ee({}, t, {
              value: ce
            }),
            warn: ee({}, t, {
              value: ie
            }),
            error: ee({}, t, {
              value: be
            }),
            group: ee({}, t, {
              value: me
            }),
            groupCollapsed: ee({}, t, {
              value: Se
            }),
            groupEnd: ee({}, t, {
              value: w
            })
          });
        }
        K < 0 && S("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var a = N.ReactCurrentDispatcher, D;
    function f(t, u, m) {
      {
        if (D === void 0)
          try {
            throw Error();
          } catch (z) {
            var R = z.stack.trim().match(/\n( *(at )?)/);
            D = R && R[1] || "";
          }
        return `
` + D + t;
      }
    }
    var B = !1, O;
    {
      var j = typeof WeakMap == "function" ? WeakMap : Map;
      O = new j();
    }
    function l(t, u) {
      if (!t || B)
        return "";
      {
        var m = O.get(t);
        if (m !== void 0)
          return m;
      }
      var R;
      B = !0;
      var z = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var U;
      U = a.current, a.current = null, y();
      try {
        if (u) {
          var W = function() {
            throw Error();
          };
          if (Object.defineProperty(W.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(W, []);
            } catch (Ae) {
              R = Ae;
            }
            Reflect.construct(t, [], W);
          } else {
            try {
              W.call();
            } catch (Ae) {
              R = Ae;
            }
            t.call(W.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (Ae) {
            R = Ae;
          }
          t();
        }
      } catch (Ae) {
        if (Ae && R && typeof Ae.stack == "string") {
          for (var M = Ae.stack.split(`
`), ue = R.stack.split(`
`), Q = M.length - 1, re = ue.length - 1; Q >= 1 && re >= 0 && M[Q] !== ue[re]; )
            re--;
          for (; Q >= 1 && re >= 0; Q--, re--)
            if (M[Q] !== ue[re]) {
              if (Q !== 1 || re !== 1)
                do
                  if (Q--, re--, re < 0 || M[Q] !== ue[re]) {
                    var ve = `
` + M[Q].replace(" at new ", " at ");
                    return t.displayName && ve.includes("<anonymous>") && (ve = ve.replace("<anonymous>", t.displayName)), typeof t == "function" && O.set(t, ve), ve;
                  }
                while (Q >= 1 && re >= 0);
              break;
            }
        }
      } finally {
        B = !1, a.current = U, A(), Error.prepareStackTrace = z;
      }
      var je = t ? t.displayName || t.name : "", At = je ? f(je) : "";
      return typeof t == "function" && O.set(t, At), At;
    }
    function q(t, u, m) {
      return l(t, !1);
    }
    function Z(t) {
      var u = t.prototype;
      return !!(u && u.isReactComponent);
    }
    function P(t, u, m) {
      if (t == null)
        return "";
      if (typeof t == "function")
        return l(t, Z(t));
      if (typeof t == "string")
        return f(t);
      switch (t) {
        case b:
          return f("Suspense");
        case g:
          return f("SuspenseList");
      }
      if (typeof t == "object")
        switch (t.$$typeof) {
          case h:
            return q(t.render);
          case E:
            return P(t.type, u, m);
          case L: {
            var R = t, z = R._payload, U = R._init;
            try {
              return P(U(z), u, m);
            } catch {
            }
          }
        }
      return "";
    }
    var J = Object.prototype.hasOwnProperty, $ = {}, te = N.ReactDebugCurrentFrame;
    function Ee(t) {
      if (t) {
        var u = t._owner, m = P(t.type, t._source, u ? u.type : null);
        te.setExtraStackFrame(m);
      } else
        te.setExtraStackFrame(null);
    }
    function ne(t, u, m, R, z) {
      {
        var U = Function.call.bind(J);
        for (var W in t)
          if (U(t, W)) {
            var M = void 0;
            try {
              if (typeof t[W] != "function") {
                var ue = Error((R || "React class") + ": " + m + " type `" + W + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof t[W] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw ue.name = "Invariant Violation", ue;
              }
              M = t[W](u, W, R, m, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (Q) {
              M = Q;
            }
            M && !(M instanceof Error) && (Ee(z), S("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", R || "React class", m, W, typeof M), Ee(null)), M instanceof Error && !(M.message in $) && ($[M.message] = !0, Ee(z), S("Failed %s type: %s", m, M.message), Ee(null));
          }
      }
    }
    var Ne = Array.isArray;
    function Re(t) {
      return Ne(t);
    }
    function fe(t) {
      {
        var u = typeof Symbol == "function" && Symbol.toStringTag, m = u && t[Symbol.toStringTag] || t.constructor.name || "Object";
        return m;
      }
    }
    function _(t) {
      try {
        return G(t), !1;
      } catch {
        return !0;
      }
    }
    function G(t) {
      return "" + t;
    }
    function De(t) {
      if (_(t))
        return S("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", fe(t)), G(t);
    }
    var xe = N.ReactCurrentOwner, Ce = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, wt, bt, rt;
    rt = {};
    function nr(t) {
      if (J.call(t, "ref")) {
        var u = Object.getOwnPropertyDescriptor(t, "ref").get;
        if (u && u.isReactWarning)
          return !1;
      }
      return t.ref !== void 0;
    }
    function or(t) {
      if (J.call(t, "key")) {
        var u = Object.getOwnPropertyDescriptor(t, "key").get;
        if (u && u.isReactWarning)
          return !1;
      }
      return t.key !== void 0;
    }
    function ar(t, u) {
      if (typeof t.ref == "string" && xe.current && u && xe.current.stateNode !== u) {
        var m = H(xe.current.type);
        rt[m] || (S('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', H(xe.current.type), t.ref), rt[m] = !0);
      }
    }
    function ir(t, u) {
      {
        var m = function() {
          wt || (wt = !0, S("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", u));
        };
        m.isReactWarning = !0, Object.defineProperty(t, "key", {
          get: m,
          configurable: !0
        });
      }
    }
    function sr(t, u) {
      {
        var m = function() {
          bt || (bt = !0, S("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", u));
        };
        m.isReactWarning = !0, Object.defineProperty(t, "ref", {
          get: m,
          configurable: !0
        });
      }
    }
    var cr = function(t, u, m, R, z, U, W) {
      var M = {
        $$typeof: n,
        type: t,
        key: u,
        ref: m,
        props: W,
        _owner: U
      };
      return M._store = {}, Object.defineProperty(M._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(M, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: R
      }), Object.defineProperty(M, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: z
      }), Object.freeze && (Object.freeze(M.props), Object.freeze(M)), M;
    };
    function lr(t, u, m, R, z) {
      {
        var U, W = {}, M = null, ue = null;
        m !== void 0 && (De(m), M = "" + m), or(u) && (De(u.key), M = "" + u.key), nr(u) && (ue = u.ref, ar(u, z));
        for (U in u)
          J.call(u, U) && !Ce.hasOwnProperty(U) && (W[U] = u[U]);
        if (t && t.defaultProps) {
          var Q = t.defaultProps;
          for (U in Q)
            W[U] === void 0 && (W[U] = Q[U]);
        }
        if (M || ue) {
          var re = typeof t == "function" ? t.displayName || t.name || "Unknown" : t;
          M && ir(W, re), ue && sr(W, re);
        }
        return cr(t, M, ue, z, R, xe.current, W);
      }
    }
    var nt = N.ReactCurrentOwner, St = N.ReactDebugCurrentFrame;
    function Me(t) {
      if (t) {
        var u = t._owner, m = P(t.type, t._source, u ? u.type : null);
        St.setExtraStackFrame(m);
      } else
        St.setExtraStackFrame(null);
    }
    var ot;
    ot = !1;
    function at(t) {
      return typeof t == "object" && t !== null && t.$$typeof === n;
    }
    function Et() {
      {
        if (nt.current) {
          var t = H(nt.current.type);
          if (t)
            return `

Check the render method of \`` + t + "`.";
        }
        return "";
      }
    }
    function ur(t) {
      {
        if (t !== void 0) {
          var u = t.fileName.replace(/^.*[\\\/]/, ""), m = t.lineNumber;
          return `

Check your code at ` + u + ":" + m + ".";
        }
        return "";
      }
    }
    var Ct = {};
    function dr(t) {
      {
        var u = Et();
        if (!u) {
          var m = typeof t == "string" ? t : t.displayName || t.name;
          m && (u = `

Check the top-level render call using <` + m + ">.");
        }
        return u;
      }
    }
    function _t(t, u) {
      {
        if (!t._store || t._store.validated || t.key != null)
          return;
        t._store.validated = !0;
        var m = dr(u);
        if (Ct[m])
          return;
        Ct[m] = !0;
        var R = "";
        t && t._owner && t._owner !== nt.current && (R = " It was passed a child from " + H(t._owner.type) + "."), Me(t), S('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', m, R), Me(null);
      }
    }
    function Rt(t, u) {
      {
        if (typeof t != "object")
          return;
        if (Re(t))
          for (var m = 0; m < t.length; m++) {
            var R = t[m];
            at(R) && _t(R, u);
          }
        else if (at(t))
          t._store && (t._store.validated = !0);
        else if (t) {
          var z = k(t);
          if (typeof z == "function" && z !== t.entries)
            for (var U = z.call(t), W; !(W = U.next()).done; )
              at(W.value) && _t(W.value, u);
        }
      }
    }
    function fr(t) {
      {
        var u = t.type;
        if (u == null || typeof u == "string")
          return;
        var m;
        if (typeof u == "function")
          m = u.propTypes;
        else if (typeof u == "object" && (u.$$typeof === h || u.$$typeof === E))
          m = u.propTypes;
        else
          return;
        if (m) {
          var R = H(u);
          ne(m, t.props, "prop", R, t);
        } else if (u.PropTypes !== void 0 && !ot) {
          ot = !0;
          var z = H(u);
          S("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", z || "Unknown");
        }
        typeof u.getDefaultProps == "function" && !u.getDefaultProps.isReactClassApproved && S("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function pr(t) {
      {
        for (var u = Object.keys(t.props), m = 0; m < u.length; m++) {
          var R = u[m];
          if (R !== "children" && R !== "key") {
            Me(t), S("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", R), Me(null);
            break;
          }
        }
        t.ref !== null && (Me(t), S("Invalid attribute `ref` supplied to `React.Fragment`."), Me(null));
      }
    }
    function xt(t, u, m, R, z, U) {
      {
        var W = he(t);
        if (!W) {
          var M = "";
          (t === void 0 || typeof t == "object" && t !== null && Object.keys(t).length === 0) && (M += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var ue = ur(z);
          ue ? M += ue : M += Et();
          var Q;
          t === null ? Q = "null" : Re(t) ? Q = "array" : t !== void 0 && t.$$typeof === n ? (Q = "<" + (H(t.type) || "Unknown") + " />", M = " Did you accidentally export a JSX literal instead of a component?") : Q = typeof t, S("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", Q, M);
        }
        var re = lr(t, u, m, z, U);
        if (re == null)
          return re;
        if (W) {
          var ve = u.children;
          if (ve !== void 0)
            if (R)
              if (Re(ve)) {
                for (var je = 0; je < ve.length; je++)
                  Rt(ve[je], t);
                Object.freeze && Object.freeze(ve);
              } else
                S("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Rt(ve, t);
        }
        return t === o ? pr(re) : fr(re), re;
      }
    }
    function hr(t, u, m) {
      return xt(t, u, m, !0);
    }
    function mr(t, u, m) {
      return xt(t, u, m, !1);
    }
    var vr = mr, gr = hr;
    Ye.Fragment = o, Ye.jsx = vr, Ye.jsxs = gr;
  }()), Ye;
}
(function(e) {
  process.env.NODE_ENV === "production" ? e.exports = Sn() : e.exports = En();
})(tt);
const Ze = tt.exports.Fragment, X = tt.exports.jsx, Ue = tt.exports.jsxs, Cn = ge.svg.withConfig({
  displayName: "MeshLogo__StyledLogo",
  componentId: "sc-1fo8pi5-0"
})({
  height: "4rem",
  padding: "0.5rem"
}), _n = () => /* @__PURE__ */ X(Cn, {
  fill: "currentColor",
  viewBox: "0 0 300 200",
  xmlns: "http://www.w3.org/2000/svg",
  children: /* @__PURE__ */ X("path", {
    d: "m289 127-45-60-45-60c-.9-1.3-2.4-2-4-2s-3.1.7-4 2l-37 49.3c-2 2.7-6 2.7-8 0l-37-49.3c-.9-1.3-2.4-2-4-2s-3.1.7-4 2l-45 60-45 60c-1.3 1.8-1.3 4.2 0 6l45 60c.9 1.3 2.4 2 4 2s3.1-.7 4-2l37-49.3c2-2.7 6-2.7 8 0l37 49.3c.9 1.3 2.4 2 4 2s3.1-.7 4-2l37-49.3c2-2.7 6-2.7 8 0l37 49.3c.9 1.3 2.4 2 4 2s3.1-.7 4-2l45-60c1.3-1.8 1.3-4.2 0-6zm-90-103.3 32.5 43.3c1.3 1.8 1.3 4.2 0 6l-32.5 43.3c-2 2.7-6 2.7-8 0l-32.5-43.3c-1.3-1.8-1.3-4.2 0-6l32.5-43.3c2-2.7 6-2.7 8 0zm-90 0 32.5 43.3c1.3 1.8 1.3 4.2 0 6l-32.5 43.3c-2 2.7-6 2.7-8 0l-32.5-43.3c-1.3-1.8-1.3-4.2 0-6l32.5-43.3c2-2.7 6-2.7 8 0zm-53 152.6-32.5-43.3c-1.3-1.8-1.3-4.2 0-6l32.5-43.3c2-2.7 6-2.7 8 0l32.5 43.3c1.3 1.8 1.3 4.2 0 6l-32.5 43.3c-2 2.7-6 2.7-8 0zm90 0-32.5-43.3c-1.3-1.8-1.3-4.2 0-6l32.5-43.3c2-2.7 6-2.7 8 0l32.5 43.3c1.3 1.8 1.3 4.2 0 6l-32.5 43.3c-2 2.7-6 2.7-8 0zm90 0-32.5-43.3c-1.3-1.8-1.3-4.2 0-6l32.5-43.3c2-2.7 6-2.7 8 0l32.5 43.3c1.3 1.8 1.3 4.2 0 6l-32.5 43.3c-2 2.7-6 2.7-8 0z"
  })
}), Rn = ge.a.withConfig({
  displayName: "MeshBadge__StyledLink",
  componentId: "sc-1maqagq-0"
})({
  display: "flex",
  maxWidth: "fit-content",
  flexDirection: "column",
  alignItems: "center",
  borderRadius: "0.375rem",
  borderWidth: "1px",
  borderStyle: "solid",
  borderColor: "currentColor",
  padding: "0.25rem",
  fontSize: "1.25rem",
  lineHeight: "1.75rem",
  fontWeight: "600",
  "--tw-text-opacity": "1",
  color: "rgb(68 64 60 / var(--tw-text-opacity))",
  textDecorationLine: "none"
}), Bn = ({
  dark: e = !1
}) => /* @__PURE__ */ Ue(Rn, {
  style: {
    color: e ? "#EEEEEE" : "#111111",
    backgroundColor: e ? "#111111" : "#EEEEEE"
  },
  href: "https://mesh.martify.io/",
  rel: "noopener noreferrer",
  target: "_blank",
  children: [/* @__PURE__ */ X(_n, {}), "Mesh"]
}), Ie = {
  walletName: "",
  walletInstance: {}
}, xn = () => {
  const [e, n] = de(void 0), [r, o] = de(!1), [i, s] = de(Ie.walletInstance), [c, p] = de(Ie.walletName), h = ct(async (g) => {
    o(!0);
    try {
      const E = await Ut.enable(g);
      s(E), p(g), n(void 0);
    } catch (E) {
      n(E), console.error(E);
    }
    o(!1);
  }, []), b = ct(() => {
    p(Ie.walletName), s(Ie.walletInstance);
  }, []);
  return {
    hasConnectedWallet: Ie.walletName !== c,
    connectedWalletInstance: i,
    connectedWalletName: c,
    connectingWallet: r,
    connectWallet: h,
    disconnect: b,
    error: e
  };
}, Pe = br({
  hasConnectedWallet: !1,
  connectedWalletInstance: Ie.walletInstance,
  connectedWalletName: Ie.walletName,
  connectingWallet: !1
}), Un = ({
  children: e
}) => {
  const n = xn();
  return /* @__PURE__ */ X(Pe.Provider, {
    value: n,
    children: e
  });
}, Hn = (e = 0) => {
  const [n, r] = de(), {
    hasConnectedWallet: o,
    connectedWalletName: i,
    connectedWalletInstance: s
  } = _e(Pe);
  return He(() => {
    o && s.getUsedAddresses().then((c) => {
      c[e] && r(c[e]);
    });
  }, [e, i]), n;
}, Gn = () => {
  const [e, n] = de(), {
    hasConnectedWallet: r,
    connectedWalletName: o,
    connectedWalletInstance: i
  } = _e(Pe);
  return He(() => {
    r && i.getAssets().then(n);
  }, [o]), e;
}, rr = () => {
  const [e, n] = de([]);
  return He(() => {
    n(Ut.getInstalledWallets());
  }, []), e;
}, An = () => {
  const [e, n] = de(), {
    hasConnectedWallet: r,
    connectedWalletName: o,
    connectedWalletInstance: i
  } = _e(Pe);
  return He(() => {
    r && i.getLovelace().then(n);
  }, [o]), e;
}, qn = () => {
  const [e, n] = de(), {
    hasConnectedWallet: r,
    connectedWalletName: o,
    connectedWalletInstance: i
  } = _e(Pe);
  return He(() => {
    r && i.getNetworkId().then(n);
  }, [o]), e;
}, kn = () => {
  const {
    hasConnectedWallet: e,
    connectedWalletName: n,
    connectedWalletInstance: r,
    connectingWallet: o,
    connectWallet: i,
    disconnect: s,
    error: c
  } = _e(Pe);
  if (i === void 0 || s === void 0)
    throw new Error("Can't call useWallet outside of the WalletProvider context");
  return {
    name: n,
    connecting: o,
    connected: e,
    wallet: r,
    connect: i,
    disconnect: s,
    error: c
  };
}, Xn = () => {
  const [e, n] = de(), [r, o] = de(), [i, s] = de(!1), {
    hasConnectedWallet: c,
    connectedWalletInstance: p
  } = _e(Pe), h = ct(async (b) => {
    s(!0);
    try {
      if (c) {
        const g = await p.submitTx(b);
        n(void 0), o(g);
      }
      throw new Error("Please make sure to connect a wallet before calling useWalletSubmit");
    } catch (g) {
      n(g), console.error(g);
    }
    s(!1);
  }, []);
  return {
    error: e,
    result: r,
    submitting: i,
    submitTx: h
  };
}, Kn = (e) => {
  const {
    era: n,
    parameters: r
  } = e, {
    hasConnectedWallet: o,
    connectedWalletInstance: i
  } = _e(Pe), [s] = de(() => {
    if (o)
      return new Sr({
        initiator: i,
        parameters: r,
        era: n
      });
    throw new Error("Please make sure to connect a wallet before calling useWalletTx");
  });
  return s;
}, Tn = ge.svg.withConfig({
  displayName: "CheckMark__StyledCheckMark",
  componentId: "sc-1h0hg7n-0"
})({
  margin: "0.5rem",
  height: "1.5rem"
}), Pn = () => /* @__PURE__ */ X(Tn, {
  fill: "currentColor",
  viewBox: "0 0 512 512",
  xmlns: "http://www.w3.org/2000/svg",
  children: /* @__PURE__ */ X("path", {
    d: "M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
  })
}), On = ge.div.withConfig({
  displayName: "MenuItem__StyledItem",
  componentId: "sc-1r28dtf-0"
})({
  display: "flex",
  cursor: "pointer",
  alignItems: "center",
  paddingLeft: "1rem",
  paddingRight: "1rem",
  paddingTop: "0.5rem",
  paddingBottom: "0.5rem",
  opacity: "0.8",
  ":hover": {
    opacity: "1"
  }
}), In = ge.img.withConfig({
  displayName: "MenuItem__StyledIcon",
  componentId: "sc-1r28dtf-1"
})({
  margin: "0.25rem",
  height: "2rem",
  paddingRight: "0.5rem"
}), Nn = ge.span.withConfig({
  displayName: "MenuItem__StyledName",
  componentId: "sc-1r28dtf-2"
})({
  fontSize: "1.25rem",
  lineHeight: "1.75rem",
  fontWeight: "400"
}), Dn = ge.span.withConfig({
  displayName: "MenuItem__StyledActive",
  componentId: "sc-1r28dtf-3"
})({
  marginLeft: "auto"
}), Mn = ({
  icon: e,
  name: n,
  connect: r,
  active: o
}) => /* @__PURE__ */ Ue(On, {
  onClick: r,
  children: [/* @__PURE__ */ X(In, {
    src: e
  }), /* @__PURE__ */ X(Nn, {
    children: n.at(0).toUpperCase() + n.slice(1).toLowerCase()
  }), o && /* @__PURE__ */ X(Dn, {
    children: /* @__PURE__ */ X(Pn, {})
  })]
}), jn = ge.svg.withConfig({
  displayName: "ChevronDown__StyledChevronDown",
  componentId: "sc-eon6h4-0"
})({
  margin: "0.5rem",
  height: "1.5rem"
}), Wn = () => /* @__PURE__ */ X(jn, {
  fill: "none",
  "aria-hidden": "true",
  viewBox: "0 0 24 24",
  stroke: "currentColor",
  xmlns: "http://www.w3.org/2000/svg",
  children: /* @__PURE__ */ X("path", {
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: "2",
    d: "M19 9l-7 7-7-7"
  })
}), Bt = ge.img.withConfig({
  displayName: "WalletBalance__StyledIcon",
  componentId: "sc-w1h5eb-0"
})({
  margin: "0.5rem",
  height: "1.5rem"
}), $n = ge.span.withConfig({
  displayName: "WalletBalance__StyledSmall",
  componentId: "sc-w1h5eb-1"
})({
  fontSize: "0.75rem",
  lineHeight: "1rem"
}), Ln = ({
  connected: e,
  name: n,
  connecting: r
}) => {
  const o = rr().find((s) => s.name === n), i = An();
  return e && i && o?.icon ? /* @__PURE__ */ Ue(Ze, {
    children: [/* @__PURE__ */ X(Bt, {
      src: o.icon
    }), "\u20B3 ", parseInt((parseInt(i, 10) / 1e6).toString(), 10), ".", /* @__PURE__ */ X($n, {
      children: i.substr(i.length - 6)
    })]
  }) : e && o?.icon ? /* @__PURE__ */ X(Ze, {
    children: /* @__PURE__ */ X(Bt, {
      src: o.icon
    })
  }) : r ? /* @__PURE__ */ X(Ze, {
    children: "Connecting..."
  }) : /* @__PURE__ */ Ue(Ze, {
    children: ["Connect Wallet ", /* @__PURE__ */ X(Wn, {})]
  });
}, Fn = ge.button.withConfig({
  displayName: "CardanoWallet__StyledMenuButton",
  componentId: "sc-1y5yscd-0"
})({
  display: "flex",
  width: "15rem",
  alignItems: "center",
  justifyContent: "center",
  borderTopLeftRadius: "0.375rem",
  borderTopRightRadius: "0.375rem",
  borderWidth: "1px",
  paddingLeft: "1rem",
  paddingRight: "1rem",
  paddingTop: "0.5rem",
  paddingBottom: "0.5rem",
  fontSize: "1.125rem",
  lineHeight: "1.75rem",
  fontWeight: "400",
  "--tw-shadow": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
  "--tw-shadow-colored": "0 1px 2px 0 var(--tw-shadow-color)",
  boxShadow: "var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow)"
}), zn = ge.div.withConfig({
  displayName: "CardanoWallet__StyledMenuList",
  componentId: "sc-1y5yscd-1"
})(({
  hidden: e
}) => [{
  position: "absolute",
  width: "15rem",
  borderBottomRightRadius: "0.375rem",
  borderBottomLeftRadius: "0.375rem",
  borderWidth: "1px",
  textAlign: "center",
  "--tw-shadow": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
  "--tw-shadow-colored": "0 1px 2px 0 var(--tw-shadow-color)",
  boxShadow: "var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow)",
  "--tw-backdrop-blur": "blur(8px)",
  backdropFilter: "var(--tw-backdrop-blur) var(--tw-backdrop-brightness) var(--tw-backdrop-contrast) var(--tw-backdrop-grayscale) var(--tw-backdrop-hue-rotate) var(--tw-backdrop-invert) var(--tw-backdrop-opacity) var(--tw-backdrop-saturate) var(--tw-backdrop-sepia)"
}, e && {
  display: "none"
}]), Zn = () => {
  const [e, n] = de(!0), {
    connect: r,
    connected: o,
    name: i,
    connecting: s
  } = kn(), c = rr();
  return /* @__PURE__ */ Ue("div", {
    onMouseEnter: () => n(!1),
    onMouseLeave: () => n(!0),
    children: [/* @__PURE__ */ X(Fn, {
      type: "button",
      onClick: () => n(!e),
      children: /* @__PURE__ */ X(Ln, {
        connected: o,
        name: i,
        connecting: s
      })
    }), /* @__PURE__ */ X(zn, {
      hidden: e,
      children: c.length > 0 ? c.map((p, h) => /* @__PURE__ */ X(Mn, {
        icon: p.icon,
        name: p.name,
        connect: () => {
          r(p.name), n(!e);
        },
        active: i === p.name
      }, h)) : /* @__PURE__ */ X("span", {
        children: "No Wallet Found."
      })
    })]
  });
};
export {
  Zn as CardanoWallet,
  Bn as MeshBadge,
  Un as MeshProvider,
  Pe as WalletContext,
  Hn as useAddress,
  Gn as useAssets,
  An as useLovelace,
  qn as useNetwork,
  kn as useWallet,
  rr as useWalletList,
  Xn as useWalletSubmit,
  Kn as useWalletTx
};
