// @meshsdk/react/dist/react.js
import Ye, { useRef as Sr, useDebugValue as Ot, useContext as ye, createElement as Cr, createContext as Er, useState as ee, useCallback as ct, useEffect as Oe } from "react";
import { BrowserWallet as qt, Transaction as _r } from "@meshsdk/core";
var qe = { exports: {} }, B = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var It;
function xr() {
  if (It)
    return B;
  It = 1;
  var e = typeof Symbol == "function" && Symbol.for, n = e ? Symbol.for("react.element") : 60103, t = e ? Symbol.for("react.portal") : 60106, o = e ? Symbol.for("react.fragment") : 60107, a = e ? Symbol.for("react.strict_mode") : 60108, c = e ? Symbol.for("react.profiler") : 60114, s = e ? Symbol.for("react.provider") : 60109, f = e ? Symbol.for("react.context") : 60110, h = e ? Symbol.for("react.async_mode") : 60111, v = e ? Symbol.for("react.concurrent_mode") : 60111, g = e ? Symbol.for("react.forward_ref") : 60112, C = e ? Symbol.for("react.suspense") : 60113, D = e ? Symbol.for("react.suspense_list") : 60120, F = e ? Symbol.for("react.memo") : 60115, E = e ? Symbol.for("react.lazy") : 60116, O = e ? Symbol.for("react.block") : 60121, x = e ? Symbol.for("react.fundamental") : 60117, I = e ? Symbol.for("react.responder") : 60118, y = e ? Symbol.for("react.scope") : 60119;
  function T(d) {
    if (typeof d == "object" && d !== null) {
      var ae = d.$$typeof;
      switch (ae) {
        case n:
          switch (d = d.type, d) {
            case h:
            case v:
            case o:
            case c:
            case a:
            case C:
              return d;
            default:
              switch (d = d && d.$$typeof, d) {
                case f:
                case g:
                case E:
                case F:
                case s:
                  return d;
                default:
                  return ae;
              }
          }
        case t:
          return ae;
      }
    }
  }
  function _(d) {
    return T(d) === v;
  }
  return B.AsyncMode = h, B.ConcurrentMode = v, B.ContextConsumer = f, B.ContextProvider = s, B.Element = n, B.ForwardRef = g, B.Fragment = o, B.Lazy = E, B.Memo = F, B.Portal = t, B.Profiler = c, B.StrictMode = a, B.Suspense = C, B.isAsyncMode = function(d) {
    return _(d) || T(d) === h;
  }, B.isConcurrentMode = _, B.isContextConsumer = function(d) {
    return T(d) === f;
  }, B.isContextProvider = function(d) {
    return T(d) === s;
  }, B.isElement = function(d) {
    return typeof d == "object" && d !== null && d.$$typeof === n;
  }, B.isForwardRef = function(d) {
    return T(d) === g;
  }, B.isFragment = function(d) {
    return T(d) === o;
  }, B.isLazy = function(d) {
    return T(d) === E;
  }, B.isMemo = function(d) {
    return T(d) === F;
  }, B.isPortal = function(d) {
    return T(d) === t;
  }, B.isProfiler = function(d) {
    return T(d) === c;
  }, B.isStrictMode = function(d) {
    return T(d) === a;
  }, B.isSuspense = function(d) {
    return T(d) === C;
  }, B.isValidElementType = function(d) {
    return typeof d == "string" || typeof d == "function" || d === o || d === v || d === c || d === a || d === C || d === D || typeof d == "object" && d !== null && (d.$$typeof === E || d.$$typeof === F || d.$$typeof === s || d.$$typeof === f || d.$$typeof === g || d.$$typeof === x || d.$$typeof === I || d.$$typeof === y || d.$$typeof === O);
  }, B.typeOf = T, B;
}
var V = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Nt;
function Rr() {
  return Nt || (Nt = 1, process.env.NODE_ENV !== "production" && function() {
    var e = typeof Symbol == "function" && Symbol.for, n = e ? Symbol.for("react.element") : 60103, t = e ? Symbol.for("react.portal") : 60106, o = e ? Symbol.for("react.fragment") : 60107, a = e ? Symbol.for("react.strict_mode") : 60108, c = e ? Symbol.for("react.profiler") : 60114, s = e ? Symbol.for("react.provider") : 60109, f = e ? Symbol.for("react.context") : 60110, h = e ? Symbol.for("react.async_mode") : 60111, v = e ? Symbol.for("react.concurrent_mode") : 60111, g = e ? Symbol.for("react.forward_ref") : 60112, C = e ? Symbol.for("react.suspense") : 60113, D = e ? Symbol.for("react.suspense_list") : 60120, F = e ? Symbol.for("react.memo") : 60115, E = e ? Symbol.for("react.lazy") : 60116, O = e ? Symbol.for("react.block") : 60121, x = e ? Symbol.for("react.fundamental") : 60117, I = e ? Symbol.for("react.responder") : 60118, y = e ? Symbol.for("react.scope") : 60119;
    function T(l) {
      return typeof l == "string" || typeof l == "function" || l === o || l === v || l === c || l === a || l === C || l === D || typeof l == "object" && l !== null && (l.$$typeof === E || l.$$typeof === F || l.$$typeof === s || l.$$typeof === f || l.$$typeof === g || l.$$typeof === x || l.$$typeof === I || l.$$typeof === y || l.$$typeof === O);
    }
    function _(l) {
      if (typeof l == "object" && l !== null) {
        var X = l.$$typeof;
        switch (X) {
          case n:
            var Z = l.type;
            switch (Z) {
              case h:
              case v:
              case o:
              case c:
              case a:
              case C:
                return Z;
              default:
                var P = Z && Z.$$typeof;
                switch (P) {
                  case f:
                  case g:
                  case E:
                  case F:
                  case s:
                    return P;
                  default:
                    return X;
                }
            }
          case t:
            return X;
        }
      }
    }
    var d = h, ae = v, he = f, ce = s, ue = n, me = g, we = o, ie = E, G = F, te = t, K = c, be = a, le = C, se = !1;
    function Se(l) {
      return se || (se = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), ve(l) || _(l) === h;
    }
    function ve(l) {
      return _(l) === v;
    }
    function Ce(l) {
      return _(l) === f;
    }
    function S(l) {
      return _(l) === s;
    }
    function w(l) {
      return typeof l == "object" && l !== null && l.$$typeof === n;
    }
    function b(l) {
      return _(l) === g;
    }
    function A(l) {
      return _(l) === o;
    }
    function i(l) {
      return _(l) === E;
    }
    function j(l) {
      return _(l) === F;
    }
    function p(l) {
      return _(l) === t;
    }
    function U(l) {
      return _(l) === c;
    }
    function M(l) {
      return _(l) === a;
    }
    function $(l) {
      return _(l) === C;
    }
    V.AsyncMode = d, V.ConcurrentMode = ae, V.ContextConsumer = he, V.ContextProvider = ce, V.Element = ue, V.ForwardRef = me, V.Fragment = we, V.Lazy = ie, V.Memo = G, V.Portal = te, V.Profiler = K, V.StrictMode = be, V.Suspense = le, V.isAsyncMode = Se, V.isConcurrentMode = ve, V.isContextConsumer = Ce, V.isContextProvider = S, V.isElement = w, V.isForwardRef = b, V.isFragment = A, V.isLazy = i, V.isMemo = j, V.isPortal = p, V.isProfiler = U, V.isStrictMode = M, V.isSuspense = $, V.isValidElementType = T, V.typeOf = _;
  }()), V;
}
(function(e) {
  process.env.NODE_ENV === "production" ? e.exports = xr() : e.exports = Rr();
})(qe);
function kr(e) {
  function n(S, w, b, A, i) {
    for (var j = 0, p = 0, U = 0, M = 0, $, l, X = 0, Z = 0, P, J = P = $ = 0, z = 0, re = 0, Ee = 0, oe = 0, je = b.length, xe = je - 1, pe, R = "", q = "", We = "", Re = "", _e; z < je; ) {
      if (l = b.charCodeAt(z), z === xe && p + M + U + j !== 0 && (p !== 0 && (l = p === 47 ? 10 : 47), M = U = j = 0, je++, xe++), p + M + U + j === 0) {
        if (z === xe && (0 < re && (R = R.replace(D, "")), 0 < R.trim().length)) {
          switch (l) {
            case 32:
            case 9:
            case 59:
            case 13:
            case 10:
              break;
            default:
              R += b.charAt(z);
          }
          l = 59;
        }
        switch (l) {
          case 123:
            for (R = R.trim(), $ = R.charCodeAt(0), P = 1, oe = ++z; z < je; ) {
              switch (l = b.charCodeAt(z)) {
                case 123:
                  P++;
                  break;
                case 125:
                  P--;
                  break;
                case 47:
                  switch (l = b.charCodeAt(z + 1)) {
                    case 42:
                    case 47:
                      e: {
                        for (J = z + 1; J < xe; ++J)
                          switch (b.charCodeAt(J)) {
                            case 47:
                              if (l === 42 && b.charCodeAt(J - 1) === 42 && z + 2 !== J) {
                                z = J + 1;
                                break e;
                              }
                              break;
                            case 10:
                              if (l === 47) {
                                z = J + 1;
                                break e;
                              }
                          }
                        z = J;
                      }
                  }
                  break;
                case 91:
                  l++;
                case 40:
                  l++;
                case 34:
                case 39:
                  for (; z++ < xe && b.charCodeAt(z) !== l; )
                    ;
              }
              if (P === 0)
                break;
              z++;
            }
            switch (P = b.substring(oe, z), $ === 0 && ($ = (R = R.replace(C, "").trim()).charCodeAt(0)), $) {
              case 64:
                switch (0 < re && (R = R.replace(D, "")), l = R.charCodeAt(1), l) {
                  case 100:
                  case 109:
                  case 115:
                  case 45:
                    re = w;
                    break;
                  default:
                    re = be;
                }
                if (P = n(w, re, P, l, i + 1), oe = P.length, 0 < se && (re = t(be, R, Ee), _e = f(3, P, re, w, G, ie, oe, l, i, A), R = re.join(""), _e !== void 0 && (oe = (P = _e.trim()).length) === 0 && (l = 0, P = "")), 0 < oe)
                  switch (l) {
                    case 115:
                      R = R.replace(ae, s);
                    case 100:
                    case 109:
                    case 45:
                      P = R + "{" + P + "}";
                      break;
                    case 107:
                      R = R.replace(y, "$1 $2"), P = R + "{" + P + "}", P = K === 1 || K === 2 && c("@" + P, 3) ? "@-webkit-" + P + "@" + P : "@" + P;
                      break;
                    default:
                      P = R + P, A === 112 && (P = (q += P, ""));
                  }
                else
                  P = "";
                break;
              default:
                P = n(w, t(w, R, Ee), P, A, i + 1);
            }
            We += P, P = Ee = re = J = $ = 0, R = "", l = b.charCodeAt(++z);
            break;
          case 125:
          case 59:
            if (R = (0 < re ? R.replace(D, "") : R).trim(), 1 < (oe = R.length))
              switch (J === 0 && ($ = R.charCodeAt(0), $ === 45 || 96 < $ && 123 > $) && (oe = (R = R.replace(" ", ":")).length), 0 < se && (_e = f(1, R, w, S, G, ie, q.length, A, i, A)) !== void 0 && (oe = (R = _e.trim()).length) === 0 && (R = "\0\0"), $ = R.charCodeAt(0), l = R.charCodeAt(1), $) {
                case 0:
                  break;
                case 64:
                  if (l === 105 || l === 99) {
                    Re += R + b.charAt(z);
                    break;
                  }
                default:
                  R.charCodeAt(oe - 1) !== 58 && (q += a(R, $, l, R.charCodeAt(2)));
              }
            Ee = re = J = $ = 0, R = "", l = b.charCodeAt(++z);
        }
      }
      switch (l) {
        case 13:
        case 10:
          p === 47 ? p = 0 : 1 + $ === 0 && A !== 107 && 0 < R.length && (re = 1, R += "\0"), 0 < se * ve && f(0, R, w, S, G, ie, q.length, A, i, A), ie = 1, G++;
          break;
        case 59:
        case 125:
          if (p + M + U + j === 0) {
            ie++;
            break;
          }
        default:
          switch (ie++, pe = b.charAt(z), l) {
            case 9:
            case 32:
              if (M + j + p === 0)
                switch (X) {
                  case 44:
                  case 58:
                  case 9:
                  case 32:
                    pe = "";
                    break;
                  default:
                    l !== 32 && (pe = " ");
                }
              break;
            case 0:
              pe = "\\0";
              break;
            case 12:
              pe = "\\f";
              break;
            case 11:
              pe = "\\v";
              break;
            case 38:
              M + p + j === 0 && (re = Ee = 1, pe = "\f" + pe);
              break;
            case 108:
              if (M + p + j + te === 0 && 0 < J)
                switch (z - J) {
                  case 2:
                    X === 112 && b.charCodeAt(z - 3) === 58 && (te = X);
                  case 8:
                    Z === 111 && (te = Z);
                }
              break;
            case 58:
              M + p + j === 0 && (J = z);
              break;
            case 44:
              p + U + M + j === 0 && (re = 1, pe += "\r");
              break;
            case 34:
            case 39:
              p === 0 && (M = M === l ? 0 : M === 0 ? l : M);
              break;
            case 91:
              M + p + U === 0 && j++;
              break;
            case 93:
              M + p + U === 0 && j--;
              break;
            case 41:
              M + p + j === 0 && U--;
              break;
            case 40:
              if (M + p + j === 0) {
                if ($ === 0)
                  switch (2 * X + 3 * Z) {
                    case 533:
                      break;
                    default:
                      $ = 1;
                  }
                U++;
              }
              break;
            case 64:
              p + U + M + j + J + P === 0 && (P = 1);
              break;
            case 42:
            case 47:
              if (!(0 < M + j + U))
                switch (p) {
                  case 0:
                    switch (2 * l + 3 * b.charCodeAt(z + 1)) {
                      case 235:
                        p = 47;
                        break;
                      case 220:
                        oe = z, p = 42;
                    }
                    break;
                  case 42:
                    l === 47 && X === 42 && oe + 2 !== z && (b.charCodeAt(oe + 2) === 33 && (q += b.substring(oe, z + 1)), pe = "", p = 0);
                }
          }
          p === 0 && (R += pe);
      }
      Z = X, X = l, z++;
    }
    if (oe = q.length, 0 < oe) {
      if (re = w, 0 < se && (_e = f(2, q, re, S, G, ie, oe, A, i, A), _e !== void 0 && (q = _e).length === 0))
        return Re + q + We;
      if (q = re.join(",") + "{" + q + "}", K * te !== 0) {
        switch (K !== 2 || c(q, 2) || (te = 0), te) {
          case 111:
            q = q.replace(_, ":-moz-$1") + q;
            break;
          case 112:
            q = q.replace(T, "::-webkit-input-$1") + q.replace(T, "::-moz-$1") + q.replace(T, ":-ms-input-$1") + q;
        }
        te = 0;
      }
    }
    return Re + q + We;
  }
  function t(S, w, b) {
    var A = w.trim().split(x);
    w = A;
    var i = A.length, j = S.length;
    switch (j) {
      case 0:
      case 1:
        var p = 0;
        for (S = j === 0 ? "" : S[0] + " "; p < i; ++p)
          w[p] = o(S, w[p], b).trim();
        break;
      default:
        var U = p = 0;
        for (w = []; p < i; ++p)
          for (var M = 0; M < j; ++M)
            w[U++] = o(S[M] + " ", A[p], b).trim();
    }
    return w;
  }
  function o(S, w, b) {
    var A = w.charCodeAt(0);
    switch (33 > A && (A = (w = w.trim()).charCodeAt(0)), A) {
      case 38:
        return w.replace(I, "$1" + S.trim());
      case 58:
        return S.trim() + w.replace(I, "$1" + S.trim());
      default:
        if (0 < 1 * b && 0 < w.indexOf("\f"))
          return w.replace(I, (S.charCodeAt(0) === 58 ? "" : "$1") + S.trim());
    }
    return S + w;
  }
  function a(S, w, b, A) {
    var i = S + ";", j = 2 * w + 3 * b + 4 * A;
    if (j === 944) {
      S = i.indexOf(":", 9) + 1;
      var p = i.substring(S, i.length - 1).trim();
      return p = i.substring(0, S).trim() + p + ";", K === 1 || K === 2 && c(p, 1) ? "-webkit-" + p + p : p;
    }
    if (K === 0 || K === 2 && !c(i, 1))
      return i;
    switch (j) {
      case 1015:
        return i.charCodeAt(10) === 97 ? "-webkit-" + i + i : i;
      case 951:
        return i.charCodeAt(3) === 116 ? "-webkit-" + i + i : i;
      case 963:
        return i.charCodeAt(5) === 110 ? "-webkit-" + i + i : i;
      case 1009:
        if (i.charCodeAt(4) !== 100)
          break;
      case 969:
      case 942:
        return "-webkit-" + i + i;
      case 978:
        return "-webkit-" + i + "-moz-" + i + i;
      case 1019:
      case 983:
        return "-webkit-" + i + "-moz-" + i + "-ms-" + i + i;
      case 883:
        if (i.charCodeAt(8) === 45)
          return "-webkit-" + i + i;
        if (0 < i.indexOf("image-set(", 11))
          return i.replace(we, "$1-webkit-$2") + i;
        break;
      case 932:
        if (i.charCodeAt(4) === 45)
          switch (i.charCodeAt(5)) {
            case 103:
              return "-webkit-box-" + i.replace("-grow", "") + "-webkit-" + i + "-ms-" + i.replace("grow", "positive") + i;
            case 115:
              return "-webkit-" + i + "-ms-" + i.replace("shrink", "negative") + i;
            case 98:
              return "-webkit-" + i + "-ms-" + i.replace("basis", "preferred-size") + i;
          }
        return "-webkit-" + i + "-ms-" + i + i;
      case 964:
        return "-webkit-" + i + "-ms-flex-" + i + i;
      case 1023:
        if (i.charCodeAt(8) !== 99)
          break;
        return p = i.substring(i.indexOf(":", 15)).replace("flex-", "").replace("space-between", "justify"), "-webkit-box-pack" + p + "-webkit-" + i + "-ms-flex-pack" + p + i;
      case 1005:
        return E.test(i) ? i.replace(F, ":-webkit-") + i.replace(F, ":-moz-") + i : i;
      case 1e3:
        switch (p = i.substring(13).trim(), w = p.indexOf("-") + 1, p.charCodeAt(0) + p.charCodeAt(w)) {
          case 226:
            p = i.replace(d, "tb");
            break;
          case 232:
            p = i.replace(d, "tb-rl");
            break;
          case 220:
            p = i.replace(d, "lr");
            break;
          default:
            return i;
        }
        return "-webkit-" + i + "-ms-" + p + i;
      case 1017:
        if (i.indexOf("sticky", 9) === -1)
          break;
      case 975:
        switch (w = (i = S).length - 10, p = (i.charCodeAt(w) === 33 ? i.substring(0, w) : i).substring(S.indexOf(":", 7) + 1).trim(), j = p.charCodeAt(0) + (p.charCodeAt(7) | 0)) {
          case 203:
            if (111 > p.charCodeAt(8))
              break;
          case 115:
            i = i.replace(p, "-webkit-" + p) + ";" + i;
            break;
          case 207:
          case 102:
            i = i.replace(p, "-webkit-" + (102 < j ? "inline-" : "") + "box") + ";" + i.replace(p, "-webkit-" + p) + ";" + i.replace(p, "-ms-" + p + "box") + ";" + i;
        }
        return i + ";";
      case 938:
        if (i.charCodeAt(5) === 45)
          switch (i.charCodeAt(6)) {
            case 105:
              return p = i.replace("-items", ""), "-webkit-" + i + "-webkit-box-" + p + "-ms-flex-" + p + i;
            case 115:
              return "-webkit-" + i + "-ms-flex-item-" + i.replace(ce, "") + i;
            default:
              return "-webkit-" + i + "-ms-flex-line-pack" + i.replace("align-content", "").replace(ce, "") + i;
          }
        break;
      case 973:
      case 989:
        if (i.charCodeAt(3) !== 45 || i.charCodeAt(4) === 122)
          break;
      case 931:
      case 953:
        if (me.test(S) === !0)
          return (p = S.substring(S.indexOf(":") + 1)).charCodeAt(0) === 115 ? a(S.replace("stretch", "fill-available"), w, b, A).replace(":fill-available", ":stretch") : i.replace(p, "-webkit-" + p) + i.replace(p, "-moz-" + p.replace("fill-", "")) + i;
        break;
      case 962:
        if (i = "-webkit-" + i + (i.charCodeAt(5) === 102 ? "-ms-" + i : "") + i, b + A === 211 && i.charCodeAt(13) === 105 && 0 < i.indexOf("transform", 10))
          return i.substring(0, i.indexOf(";", 27) + 1).replace(O, "$1-webkit-$2") + i;
    }
    return i;
  }
  function c(S, w) {
    var b = S.indexOf(w === 1 ? ":" : "{"), A = S.substring(0, w !== 3 ? b : 10);
    return b = S.substring(b + 1, S.length - 1), Se(w !== 2 ? A : A.replace(ue, "$1"), b, w);
  }
  function s(S, w) {
    var b = a(w, w.charCodeAt(0), w.charCodeAt(1), w.charCodeAt(2));
    return b !== w + ";" ? b.replace(he, " or ($1)").substring(4) : "(" + w + ")";
  }
  function f(S, w, b, A, i, j, p, U, M, $) {
    for (var l = 0, X = w, Z; l < se; ++l)
      switch (Z = le[l].call(g, S, X, b, A, i, j, p, U, M, $)) {
        case void 0:
        case !1:
        case !0:
        case null:
          break;
        default:
          X = Z;
      }
    if (X !== w)
      return X;
  }
  function h(S) {
    switch (S) {
      case void 0:
      case null:
        se = le.length = 0;
        break;
      default:
        if (typeof S == "function")
          le[se++] = S;
        else if (typeof S == "object")
          for (var w = 0, b = S.length; w < b; ++w)
            h(S[w]);
        else
          ve = !!S | 0;
    }
    return h;
  }
  function v(S) {
    return S = S.prefix, S !== void 0 && (Se = null, S ? typeof S != "function" ? K = 1 : (K = 2, Se = S) : K = 0), v;
  }
  function g(S, w) {
    var b = S;
    if (33 > b.charCodeAt(0) && (b = b.trim()), Ce = b, b = [Ce], 0 < se) {
      var A = f(-1, w, b, b, G, ie, 0, 0, 0, 0);
      A !== void 0 && typeof A == "string" && (w = A);
    }
    var i = n(be, b, w, 0, 0);
    return 0 < se && (A = f(-2, i, b, b, G, ie, i.length, 0, 0, 0), A !== void 0 && (i = A)), Ce = "", te = 0, ie = G = 1, i;
  }
  var C = /^\0+/g, D = /[\0\r\f]/g, F = /: */g, E = /zoo|gra/, O = /([,: ])(transform)/g, x = /,\r+?/g, I = /([\t\r\n ])*\f?&/g, y = /@(k\w+)\s*(\S*)\s*/, T = /::(place)/g, _ = /:(read-only)/g, d = /[svh]\w+-[tblr]{2}/, ae = /\(\s*(.*)\s*\)/g, he = /([\s\S]*?);/g, ce = /-self|flex-/g, ue = /[^]*?(:[rp][el]a[\w-]+)[^]*/, me = /stretch|:\s*\w+\-(?:conte|avail)/, we = /([^-])(image-set\()/, ie = 1, G = 1, te = 0, K = 1, be = [], le = [], se = 0, Se = null, ve = 0, Ce = "";
  return g.use = h, g.set = v, e !== void 0 && v(e), g;
}
var Ar = {
  animationIterationCount: 1,
  borderImageOutset: 1,
  borderImageSlice: 1,
  borderImageWidth: 1,
  boxFlex: 1,
  boxFlexGroup: 1,
  boxOrdinalGroup: 1,
  columnCount: 1,
  columns: 1,
  flex: 1,
  flexGrow: 1,
  flexPositive: 1,
  flexShrink: 1,
  flexNegative: 1,
  flexOrder: 1,
  gridRow: 1,
  gridRowEnd: 1,
  gridRowSpan: 1,
  gridRowStart: 1,
  gridColumn: 1,
  gridColumnEnd: 1,
  gridColumnSpan: 1,
  gridColumnStart: 1,
  msGridRow: 1,
  msGridRowSpan: 1,
  msGridColumn: 1,
  msGridColumnSpan: 1,
  fontWeight: 1,
  lineHeight: 1,
  opacity: 1,
  order: 1,
  orphans: 1,
  tabSize: 1,
  widows: 1,
  zIndex: 1,
  zoom: 1,
  WebkitLineClamp: 1,
  fillOpacity: 1,
  floodOpacity: 1,
  stopOpacity: 1,
  strokeDasharray: 1,
  strokeDashoffset: 1,
  strokeMiterlimit: 1,
  strokeOpacity: 1,
  strokeWidth: 1
};
function Tr(e) {
  var n = /* @__PURE__ */ Object.create(null);
  return function(t) {
    return n[t] === void 0 && (n[t] = e(t)), n[t];
  };
}
var Pr = /^((children|dangerouslySetInnerHTML|key|ref|autoFocus|defaultValue|defaultChecked|innerHTML|suppressContentEditableWarning|suppressHydrationWarning|valueLink|abbr|accept|acceptCharset|accessKey|action|allow|allowUserMedia|allowPaymentRequest|allowFullScreen|allowTransparency|alt|async|autoComplete|autoPlay|capture|cellPadding|cellSpacing|challenge|charSet|checked|cite|classID|className|cols|colSpan|content|contentEditable|contextMenu|controls|controlsList|coords|crossOrigin|data|dateTime|decoding|default|defer|dir|disabled|disablePictureInPicture|download|draggable|encType|enterKeyHint|form|formAction|formEncType|formMethod|formNoValidate|formTarget|frameBorder|headers|height|hidden|high|href|hrefLang|htmlFor|httpEquiv|id|inputMode|integrity|is|keyParams|keyType|kind|label|lang|list|loading|loop|low|marginHeight|marginWidth|max|maxLength|media|mediaGroup|method|min|minLength|multiple|muted|name|nonce|noValidate|open|optimum|pattern|placeholder|playsInline|poster|preload|profile|radioGroup|readOnly|referrerPolicy|rel|required|reversed|role|rows|rowSpan|sandbox|scope|scoped|scrolling|seamless|selected|shape|size|sizes|slot|span|spellCheck|src|srcDoc|srcLang|srcSet|start|step|style|summary|tabIndex|target|title|translate|type|useMap|value|width|wmode|wrap|about|datatype|inlist|prefix|property|resource|typeof|vocab|autoCapitalize|autoCorrect|autoSave|color|incremental|fallback|inert|itemProp|itemScope|itemType|itemID|itemRef|on|option|results|security|unselectable|accentHeight|accumulate|additive|alignmentBaseline|allowReorder|alphabetic|amplitude|arabicForm|ascent|attributeName|attributeType|autoReverse|azimuth|baseFrequency|baselineShift|baseProfile|bbox|begin|bias|by|calcMode|capHeight|clip|clipPathUnits|clipPath|clipRule|colorInterpolation|colorInterpolationFilters|colorProfile|colorRendering|contentScriptType|contentStyleType|cursor|cx|cy|d|decelerate|descent|diffuseConstant|direction|display|divisor|dominantBaseline|dur|dx|dy|edgeMode|elevation|enableBackground|end|exponent|externalResourcesRequired|fill|fillOpacity|fillRule|filter|filterRes|filterUnits|floodColor|floodOpacity|focusable|fontFamily|fontSize|fontSizeAdjust|fontStretch|fontStyle|fontVariant|fontWeight|format|from|fr|fx|fy|g1|g2|glyphName|glyphOrientationHorizontal|glyphOrientationVertical|glyphRef|gradientTransform|gradientUnits|hanging|horizAdvX|horizOriginX|ideographic|imageRendering|in|in2|intercept|k|k1|k2|k3|k4|kernelMatrix|kernelUnitLength|kerning|keyPoints|keySplines|keyTimes|lengthAdjust|letterSpacing|lightingColor|limitingConeAngle|local|markerEnd|markerMid|markerStart|markerHeight|markerUnits|markerWidth|mask|maskContentUnits|maskUnits|mathematical|mode|numOctaves|offset|opacity|operator|order|orient|orientation|origin|overflow|overlinePosition|overlineThickness|panose1|paintOrder|pathLength|patternContentUnits|patternTransform|patternUnits|pointerEvents|points|pointsAtX|pointsAtY|pointsAtZ|preserveAlpha|preserveAspectRatio|primitiveUnits|r|radius|refX|refY|renderingIntent|repeatCount|repeatDur|requiredExtensions|requiredFeatures|restart|result|rotate|rx|ry|scale|seed|shapeRendering|slope|spacing|specularConstant|specularExponent|speed|spreadMethod|startOffset|stdDeviation|stemh|stemv|stitchTiles|stopColor|stopOpacity|strikethroughPosition|strikethroughThickness|string|stroke|strokeDasharray|strokeDashoffset|strokeLinecap|strokeLinejoin|strokeMiterlimit|strokeOpacity|strokeWidth|surfaceScale|systemLanguage|tableValues|targetX|targetY|textAnchor|textDecoration|textRendering|textLength|to|transform|u1|u2|underlinePosition|underlineThickness|unicode|unicodeBidi|unicodeRange|unitsPerEm|vAlphabetic|vHanging|vIdeographic|vMathematical|values|vectorEffect|version|vertAdvY|vertOriginX|vertOriginY|viewBox|viewTarget|visibility|widths|wordSpacing|writingMode|x|xHeight|x1|x2|xChannelSelector|xlinkActuate|xlinkArcrole|xlinkHref|xlinkRole|xlinkShow|xlinkTitle|xlinkType|xmlBase|xmlns|xmlnsXlink|xmlLang|xmlSpace|y|y1|y2|yChannelSelector|z|zoomAndPan|for|class|autofocus)|(([Dd][Aa][Tt][Aa]|[Aa][Rr][Ii][Aa]|x)-.*))$/, Dt = /* @__PURE__ */ Tr(
  function(e) {
    return Pr.test(e) || e.charCodeAt(0) === 111 && e.charCodeAt(1) === 110 && e.charCodeAt(2) < 91;
  }
), vt = qe.exports, Or = {
  childContextTypes: !0,
  contextType: !0,
  contextTypes: !0,
  defaultProps: !0,
  displayName: !0,
  getDefaultProps: !0,
  getDerivedStateFromError: !0,
  getDerivedStateFromProps: !0,
  mixins: !0,
  propTypes: !0,
  type: !0
}, Ir = {
  name: !0,
  length: !0,
  prototype: !0,
  caller: !0,
  callee: !0,
  arguments: !0,
  arity: !0
}, Nr = {
  $$typeof: !0,
  render: !0,
  defaultProps: !0,
  displayName: !0,
  propTypes: !0
}, Xt = {
  $$typeof: !0,
  compare: !0,
  defaultProps: !0,
  displayName: !0,
  propTypes: !0,
  type: !0
}, gt = {};
gt[vt.ForwardRef] = Nr;
gt[vt.Memo] = Xt;
function Mt(e) {
  return vt.isMemo(e) ? Xt : gt[e.$$typeof] || Or;
}
var Dr = Object.defineProperty, Mr = Object.getOwnPropertyNames, jt = Object.getOwnPropertySymbols, jr = Object.getOwnPropertyDescriptor, Wr = Object.getPrototypeOf, Wt = Object.prototype;
function Kt(e, n, t) {
  if (typeof n != "string") {
    if (Wt) {
      var o = Wr(n);
      o && o !== Wt && Kt(e, o, t);
    }
    var a = Mr(n);
    jt && (a = a.concat(jt(n)));
    for (var c = Mt(e), s = Mt(n), f = 0; f < a.length; ++f) {
      var h = a[f];
      if (!Ir[h] && !(t && t[h]) && !(s && s[h]) && !(c && c[h])) {
        var v = jr(n, h);
        try {
          Dr(e, h, v);
        } catch {
        }
      }
    }
  }
  return e;
}
var $r = Kt;
function Ae() {
  return (Ae = Object.assign || function(e) {
    for (var n = 1; n < arguments.length; n++) {
      var t = arguments[n];
      for (var o in t)
        Object.prototype.hasOwnProperty.call(t, o) && (e[o] = t[o]);
    }
    return e;
  }).apply(this, arguments);
}
var $t = function(e, n) {
  for (var t = [e[0]], o = 0, a = n.length; o < a; o += 1)
    t.push(n[o], e[o + 1]);
  return t;
}, lt = function(e) {
  return e !== null && typeof e == "object" && (e.toString ? e.toString() : Object.prototype.toString.call(e)) === "[object Object]" && !qe.exports.typeOf(e);
}, Qe = Object.freeze([]), Pe = Object.freeze({});
function Ge(e) {
  return typeof e == "function";
}
function ut(e) {
  return process.env.NODE_ENV !== "production" && typeof e == "string" && e || e.displayName || e.name || "Component";
}
function yt(e) {
  return e && typeof e.styledComponentId == "string";
}
var Fe = typeof process < "u" && (process.env.REACT_APP_SC_ATTR || process.env.SC_ATTR) || "data-styled", wt = typeof window < "u" && "HTMLElement" in window, Lr = Boolean(typeof SC_DISABLE_SPEEDY == "boolean" ? SC_DISABLE_SPEEDY : typeof process < "u" && process.env.REACT_APP_SC_DISABLE_SPEEDY !== void 0 && process.env.REACT_APP_SC_DISABLE_SPEEDY !== "" ? process.env.REACT_APP_SC_DISABLE_SPEEDY !== "false" && process.env.REACT_APP_SC_DISABLE_SPEEDY : typeof process < "u" && process.env.SC_DISABLE_SPEEDY !== void 0 && process.env.SC_DISABLE_SPEEDY !== "" ? process.env.SC_DISABLE_SPEEDY !== "false" && process.env.SC_DISABLE_SPEEDY : process.env.NODE_ENV !== "production"), Fr = process.env.NODE_ENV !== "production" ? { 1: `Cannot create styled-component for component: %s.

`, 2: `Can't collect styles once you've consumed a \`ServerStyleSheet\`'s styles! \`ServerStyleSheet\` is a one off instance for each server-side render cycle.

- Are you trying to reuse it across renders?
- Are you accidentally calling collectStyles twice?

`, 3: `Streaming SSR is only supported in a Node.js environment; Please do not try to call this method in the browser.

`, 4: `The \`StyleSheetManager\` expects a valid target or sheet prop!

- Does this error occur on the client and is your target falsy?
- Does this error occur on the server and is the sheet falsy?

`, 5: `The clone method cannot be used on the client!

- Are you running in a client-like environment on the server?
- Are you trying to run SSR on the client?

`, 6: `Trying to insert a new style tag, but the given Node is unmounted!

- Are you using a custom target that isn't mounted?
- Does your document not have a valid head element?
- Have you accidentally removed a style tag manually?

`, 7: 'ThemeProvider: Please return an object from your "theme" prop function, e.g.\n\n```js\ntheme={() => ({})}\n```\n\n', 8: `ThemeProvider: Please make your "theme" prop an object.

`, 9: "Missing document `<head>`\n\n", 10: `Cannot find a StyleSheet instance. Usually this happens if there are multiple copies of styled-components loaded at once. Check out this issue for how to troubleshoot and fix the common cases where this situation can happen: https://github.com/styled-components/styled-components/issues/1941#issuecomment-417862021

`, 11: `_This error was replaced with a dev-time warning, it will be deleted for v4 final._ [createGlobalStyle] received children which will not be rendered. Please use the component without passing children elements.

`, 12: "It seems you are interpolating a keyframe declaration (%s) into an untagged string. This was supported in styled-components v3, but is not longer supported in v4 as keyframes are now injected on-demand. Please wrap your string in the css\\`\\` helper which ensures the styles are injected correctly. See https://www.styled-components.com/docs/api#css\n\n", 13: `%s is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.

`, 14: `ThemeProvider: "theme" prop is required.

`, 15: "A stylis plugin has been supplied that is not named. We need a name for each plugin to be able to prevent styling collisions between different stylis configurations within the same app. Before you pass your plugin to `<StyleSheetManager stylisPlugins={[]}>`, please make sure each plugin is uniquely-named, e.g.\n\n```js\nObject.defineProperty(importedPlugin, 'name', { value: 'some-unique-name' });\n```\n\n", 16: `Reached the limit of how many styled components may be created at group %s.
You may only create up to 1,073,741,824 components. If you're creating components dynamically,
as for instance in your render method then you may be running into this limitation.

`, 17: `CSSStyleSheet could not be found on HTMLStyleElement.
Has styled-components' style tag been unmounted or altered by another script?
` } : {};
function zr() {
  for (var e = arguments.length <= 0 ? void 0 : arguments[0], n = [], t = 1, o = arguments.length; t < o; t += 1)
    n.push(t < 0 || arguments.length <= t ? void 0 : arguments[t]);
  return n.forEach(function(a) {
    e = e.replace(/%[a-z]/, a);
  }), e;
}
function Be(e) {
  for (var n = arguments.length, t = new Array(n > 1 ? n - 1 : 0), o = 1; o < n; o++)
    t[o - 1] = arguments[o];
  throw process.env.NODE_ENV === "production" ? new Error("An error occurred. See https://git.io/JUIaE#" + e + " for more information." + (t.length > 0 ? " Args: " + t.join(", ") : "")) : new Error(zr.apply(void 0, [Fr[e]].concat(t)).trim());
}
var Yr = function() {
  function e(t) {
    this.groupSizes = new Uint32Array(512), this.length = 512, this.tag = t;
  }
  var n = e.prototype;
  return n.indexOfGroup = function(t) {
    for (var o = 0, a = 0; a < t; a++)
      o += this.groupSizes[a];
    return o;
  }, n.insertRules = function(t, o) {
    if (t >= this.groupSizes.length) {
      for (var a = this.groupSizes, c = a.length, s = c; t >= s; )
        (s <<= 1) < 0 && Be(16, "" + t);
      this.groupSizes = new Uint32Array(s), this.groupSizes.set(a), this.length = s;
      for (var f = c; f < s; f++)
        this.groupSizes[f] = 0;
    }
    for (var h = this.indexOfGroup(t + 1), v = 0, g = o.length; v < g; v++)
      this.tag.insertRule(h, o[v]) && (this.groupSizes[t]++, h++);
  }, n.clearGroup = function(t) {
    if (t < this.length) {
      var o = this.groupSizes[t], a = this.indexOfGroup(t), c = a + o;
      this.groupSizes[t] = 0;
      for (var s = a; s < c; s++)
        this.tag.deleteRule(a);
    }
  }, n.getGroup = function(t) {
    var o = "";
    if (t >= this.length || this.groupSizes[t] === 0)
      return o;
    for (var a = this.groupSizes[t], c = this.indexOfGroup(t), s = c + a, f = c; f < s; f++)
      o += this.tag.getRule(f) + `/*!sc*/
`;
    return o;
  }, e;
}(), Je = /* @__PURE__ */ new Map(), et = /* @__PURE__ */ new Map(), He = 1, Xe = function(e) {
  if (Je.has(e))
    return Je.get(e);
  for (; et.has(He); )
    He++;
  var n = He++;
  return process.env.NODE_ENV !== "production" && ((0 | n) < 0 || n > 1 << 30) && Be(16, "" + n), Je.set(e, n), et.set(n, e), n;
}, Br = function(e) {
  return et.get(e);
}, Vr = function(e, n) {
  n >= He && (He = n + 1), Je.set(e, n), et.set(n, e);
}, Ur = "style[" + Fe + '][data-styled-version="5.3.6"]', Hr = new RegExp("^" + Fe + '\\.g(\\d+)\\[id="([\\w\\d-]+)"\\].*?"([^"]*)'), Gr = function(e, n, t) {
  for (var o, a = t.split(","), c = 0, s = a.length; c < s; c++)
    (o = a[c]) && e.registerName(n, o);
}, qr = function(e, n) {
  for (var t = (n.textContent || "").split(`/*!sc*/
`), o = [], a = 0, c = t.length; a < c; a++) {
    var s = t[a].trim();
    if (s) {
      var f = s.match(Hr);
      if (f) {
        var h = 0 | parseInt(f[1], 10), v = f[2];
        h !== 0 && (Vr(v, h), Gr(e, v, f[3]), e.getTag().insertRules(h, o)), o.length = 0;
      } else
        o.push(s);
    }
  }
}, Xr = function() {
  return typeof __webpack_nonce__ < "u" ? __webpack_nonce__ : null;
}, Zt = function(e) {
  var n = document.head, t = e || n, o = document.createElement("style"), a = function(f) {
    for (var h = f.childNodes, v = h.length; v >= 0; v--) {
      var g = h[v];
      if (g && g.nodeType === 1 && g.hasAttribute(Fe))
        return g;
    }
  }(t), c = a !== void 0 ? a.nextSibling : null;
  o.setAttribute(Fe, "active"), o.setAttribute("data-styled-version", "5.3.6");
  var s = Xr();
  return s && o.setAttribute("nonce", s), t.insertBefore(o, c), o;
}, Kr = function() {
  function e(t) {
    var o = this.element = Zt(t);
    o.appendChild(document.createTextNode("")), this.sheet = function(a) {
      if (a.sheet)
        return a.sheet;
      for (var c = document.styleSheets, s = 0, f = c.length; s < f; s++) {
        var h = c[s];
        if (h.ownerNode === a)
          return h;
      }
      Be(17);
    }(o), this.length = 0;
  }
  var n = e.prototype;
  return n.insertRule = function(t, o) {
    try {
      return this.sheet.insertRule(o, t), this.length++, !0;
    } catch {
      return !1;
    }
  }, n.deleteRule = function(t) {
    this.sheet.deleteRule(t), this.length--;
  }, n.getRule = function(t) {
    var o = this.sheet.cssRules[t];
    return o !== void 0 && typeof o.cssText == "string" ? o.cssText : "";
  }, e;
}(), Zr = function() {
  function e(t) {
    var o = this.element = Zt(t);
    this.nodes = o.childNodes, this.length = 0;
  }
  var n = e.prototype;
  return n.insertRule = function(t, o) {
    if (t <= this.length && t >= 0) {
      var a = document.createTextNode(o), c = this.nodes[t];
      return this.element.insertBefore(a, c || null), this.length++, !0;
    }
    return !1;
  }, n.deleteRule = function(t) {
    this.element.removeChild(this.nodes[t]), this.length--;
  }, n.getRule = function(t) {
    return t < this.length ? this.nodes[t].textContent : "";
  }, e;
}(), Jr = function() {
  function e(t) {
    this.rules = [], this.length = 0;
  }
  var n = e.prototype;
  return n.insertRule = function(t, o) {
    return t <= this.length && (this.rules.splice(t, 0, o), this.length++, !0);
  }, n.deleteRule = function(t) {
    this.rules.splice(t, 1), this.length--;
  }, n.getRule = function(t) {
    return t < this.length ? this.rules[t] : "";
  }, e;
}(), Lt = wt, Qr = { isServer: !wt, useCSSOMInjection: !Lr }, Jt = function() {
  function e(t, o, a) {
    t === void 0 && (t = Pe), o === void 0 && (o = {}), this.options = Ae({}, Qr, {}, t), this.gs = o, this.names = new Map(a), this.server = !!t.isServer, !this.server && wt && Lt && (Lt = !1, function(c) {
      for (var s = document.querySelectorAll(Ur), f = 0, h = s.length; f < h; f++) {
        var v = s[f];
        v && v.getAttribute(Fe) !== "active" && (qr(c, v), v.parentNode && v.parentNode.removeChild(v));
      }
    }(this));
  }
  e.registerId = function(t) {
    return Xe(t);
  };
  var n = e.prototype;
  return n.reconstructWithOptions = function(t, o) {
    return o === void 0 && (o = !0), new e(Ae({}, this.options, {}, t), this.gs, o && this.names || void 0);
  }, n.allocateGSInstance = function(t) {
    return this.gs[t] = (this.gs[t] || 0) + 1;
  }, n.getTag = function() {
    return this.tag || (this.tag = (a = (o = this.options).isServer, c = o.useCSSOMInjection, s = o.target, t = a ? new Jr(s) : c ? new Kr(s) : new Zr(s), new Yr(t)));
    var t, o, a, c, s;
  }, n.hasNameForId = function(t, o) {
    return this.names.has(t) && this.names.get(t).has(o);
  }, n.registerName = function(t, o) {
    if (Xe(t), this.names.has(t))
      this.names.get(t).add(o);
    else {
      var a = /* @__PURE__ */ new Set();
      a.add(o), this.names.set(t, a);
    }
  }, n.insertRules = function(t, o, a) {
    this.registerName(t, o), this.getTag().insertRules(Xe(t), a);
  }, n.clearNames = function(t) {
    this.names.has(t) && this.names.get(t).clear();
  }, n.clearRules = function(t) {
    this.getTag().clearGroup(Xe(t)), this.clearNames(t);
  }, n.clearTag = function() {
    this.tag = void 0;
  }, n.toString = function() {
    return function(t) {
      for (var o = t.getTag(), a = o.length, c = "", s = 0; s < a; s++) {
        var f = Br(s);
        if (f !== void 0) {
          var h = t.names.get(f), v = o.getGroup(s);
          if (h && v && h.size) {
            var g = Fe + ".g" + s + '[id="' + f + '"]', C = "";
            h !== void 0 && h.forEach(function(D) {
              D.length > 0 && (C += D + ",");
            }), c += "" + v + g + '{content:"' + C + `"}/*!sc*/
`;
          }
        }
      }
      return c;
    }(this);
  }, e;
}(), en = /(a)(d)/gi, Ft = function(e) {
  return String.fromCharCode(e + (e > 25 ? 39 : 97));
};
function dt(e) {
  var n, t = "";
  for (n = Math.abs(e); n > 52; n = n / 52 | 0)
    t = Ft(n % 52) + t;
  return (Ft(n % 52) + t).replace(en, "$1-$2");
}
var Ie = function(e, n) {
  for (var t = n.length; t; )
    e = 33 * e ^ n.charCodeAt(--t);
  return e;
}, Qt = function(e) {
  return Ie(5381, e);
};
function tn(e) {
  for (var n = 0; n < e.length; n += 1) {
    var t = e[n];
    if (Ge(t) && !yt(t))
      return !1;
  }
  return !0;
}
var rn = Qt("5.3.6"), nn = function() {
  function e(n, t, o) {
    this.rules = n, this.staticRulesId = "", this.isStatic = process.env.NODE_ENV === "production" && (o === void 0 || o.isStatic) && tn(n), this.componentId = t, this.baseHash = Ie(rn, t), this.baseStyle = o, Jt.registerId(t);
  }
  return e.prototype.generateAndInjectStyles = function(n, t, o) {
    var a = this.componentId, c = [];
    if (this.baseStyle && c.push(this.baseStyle.generateAndInjectStyles(n, t, o)), this.isStatic && !o.hash)
      if (this.staticRulesId && t.hasNameForId(a, this.staticRulesId))
        c.push(this.staticRulesId);
      else {
        var s = ze(this.rules, n, t, o).join(""), f = dt(Ie(this.baseHash, s) >>> 0);
        if (!t.hasNameForId(a, f)) {
          var h = o(s, "." + f, void 0, a);
          t.insertRules(a, f, h);
        }
        c.push(f), this.staticRulesId = f;
      }
    else {
      for (var v = this.rules.length, g = Ie(this.baseHash, o.hash), C = "", D = 0; D < v; D++) {
        var F = this.rules[D];
        if (typeof F == "string")
          C += F, process.env.NODE_ENV !== "production" && (g = Ie(g, F + D));
        else if (F) {
          var E = ze(F, n, t, o), O = Array.isArray(E) ? E.join("") : E;
          g = Ie(g, O + D), C += O;
        }
      }
      if (C) {
        var x = dt(g >>> 0);
        if (!t.hasNameForId(a, x)) {
          var I = o(C, "." + x, void 0, a);
          t.insertRules(a, x, I);
        }
        c.push(x);
      }
    }
    return c.join(" ");
  }, e;
}(), on = /^\s*\/\/.*$/gm, an = [":", "[", ".", "#"];
function sn(e) {
  var n, t, o, a, c = e === void 0 ? Pe : e, s = c.options, f = s === void 0 ? Pe : s, h = c.plugins, v = h === void 0 ? Qe : h, g = new kr(f), C = [], D = function(O) {
    function x(I) {
      if (I)
        try {
          O(I + "}");
        } catch {
        }
    }
    return function(I, y, T, _, d, ae, he, ce, ue, me) {
      switch (I) {
        case 1:
          if (ue === 0 && y.charCodeAt(0) === 64)
            return O(y + ";"), "";
          break;
        case 2:
          if (ce === 0)
            return y + "/*|*/";
          break;
        case 3:
          switch (ce) {
            case 102:
            case 112:
              return O(T[0] + y), "";
            default:
              return y + (me === 0 ? "/*|*/" : "");
          }
        case -2:
          y.split("/*|*/}").forEach(x);
      }
    };
  }(function(O) {
    C.push(O);
  }), F = function(O, x, I) {
    return x === 0 && an.indexOf(I[t.length]) !== -1 || I.match(a) ? O : "." + n;
  };
  function E(O, x, I, y) {
    y === void 0 && (y = "&");
    var T = O.replace(on, ""), _ = x && I ? I + " " + x + " { " + T + " }" : T;
    return n = y, t = x, o = new RegExp("\\" + t + "\\b", "g"), a = new RegExp("(\\" + t + "\\b){2,}"), g(I || !x ? "" : x, _);
  }
  return g.use([].concat(v, [function(O, x, I) {
    O === 2 && I.length && I[0].lastIndexOf(t) > 0 && (I[0] = I[0].replace(o, F));
  }, D, function(O) {
    if (O === -2) {
      var x = C;
      return C = [], x;
    }
  }])), E.hash = v.length ? v.reduce(function(O, x) {
    return x.name || Be(15), Ie(O, x.name);
  }, 5381).toString() : "", E;
}
var er = Ye.createContext();
er.Consumer;
var tr = Ye.createContext(), cn = (tr.Consumer, new Jt()), ft = sn();
function ln() {
  return ye(er) || cn;
}
function un() {
  return ye(tr) || ft;
}
var dn = function() {
  function e(n, t) {
    var o = this;
    this.inject = function(a, c) {
      c === void 0 && (c = ft);
      var s = o.name + c.hash;
      a.hasNameForId(o.id, s) || a.insertRules(o.id, s, c(o.rules, s, "@keyframes"));
    }, this.toString = function() {
      return Be(12, String(o.name));
    }, this.name = n, this.id = "sc-keyframes-" + n, this.rules = t;
  }
  return e.prototype.getName = function(n) {
    return n === void 0 && (n = ft), this.name + n.hash;
  }, e;
}(), fn = /([A-Z])/, pn = /([A-Z])/g, hn = /^ms-/, mn = function(e) {
  return "-" + e.toLowerCase();
};
function zt(e) {
  return fn.test(e) ? e.replace(pn, mn).replace(hn, "-ms-") : e;
}
var Yt = function(e) {
  return e == null || e === !1 || e === "";
};
function ze(e, n, t, o) {
  if (Array.isArray(e)) {
    for (var a, c = [], s = 0, f = e.length; s < f; s += 1)
      (a = ze(e[s], n, t, o)) !== "" && (Array.isArray(a) ? c.push.apply(c, a) : c.push(a));
    return c;
  }
  if (Yt(e))
    return "";
  if (yt(e))
    return "." + e.styledComponentId;
  if (Ge(e)) {
    if (typeof (v = e) != "function" || v.prototype && v.prototype.isReactComponent || !n)
      return e;
    var h = e(n);
    return process.env.NODE_ENV !== "production" && qe.exports.isElement(h) && console.warn(ut(e) + " is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details."), ze(h, n, t, o);
  }
  var v;
  return e instanceof dn ? t ? (e.inject(t, o), e.getName(o)) : e : lt(e) ? function g(C, D) {
    var F, E, O = [];
    for (var x in C)
      C.hasOwnProperty(x) && !Yt(C[x]) && (Array.isArray(C[x]) && C[x].isCss || Ge(C[x]) ? O.push(zt(x) + ":", C[x], ";") : lt(C[x]) ? O.push.apply(O, g(C[x], x)) : O.push(zt(x) + ": " + (F = x, (E = C[x]) == null || typeof E == "boolean" || E === "" ? "" : typeof E != "number" || E === 0 || F in Ar ? String(E).trim() : E + "px") + ";"));
    return D ? [D + " {"].concat(O, ["}"]) : O;
  }(e) : e.toString();
}
var Bt = function(e) {
  return Array.isArray(e) && (e.isCss = !0), e;
};
function vn(e) {
  for (var n = arguments.length, t = new Array(n > 1 ? n - 1 : 0), o = 1; o < n; o++)
    t[o - 1] = arguments[o];
  return Ge(e) || lt(e) ? Bt(ze($t(Qe, [e].concat(t)))) : t.length === 0 && e.length === 1 && typeof e[0] == "string" ? e : Bt(ze($t(e, t)));
}
var Vt = /invalid hook call/i, Ke = /* @__PURE__ */ new Set(), gn = function(e, n) {
  if (process.env.NODE_ENV !== "production") {
    var t = "The component " + e + (n ? ' with the id of "' + n + '"' : "") + ` has been created dynamically.
You may see this warning because you've called styled inside another component.
To resolve this only create new StyledComponents outside of any render method and function component.`, o = console.error;
    try {
      var a = !0;
      console.error = function(c) {
        if (Vt.test(c))
          a = !1, Ke.delete(t);
        else {
          for (var s = arguments.length, f = new Array(s > 1 ? s - 1 : 0), h = 1; h < s; h++)
            f[h - 1] = arguments[h];
          o.apply(void 0, [c].concat(f));
        }
      }, Sr(), a && !Ke.has(t) && (console.warn(t), Ke.add(t));
    } catch (c) {
      Vt.test(c.message) && Ke.delete(t);
    } finally {
      console.error = o;
    }
  }
}, yn = function(e, n, t) {
  return t === void 0 && (t = Pe), e.theme !== t.theme && e.theme || n || t.theme;
}, wn = /[!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~-]+/g, bn = /(^-|-$)/g;
function it(e) {
  return e.replace(wn, "-").replace(bn, "");
}
var Sn = function(e) {
  return dt(Qt(e) >>> 0);
};
function Ze(e) {
  return typeof e == "string" && (process.env.NODE_ENV === "production" || e.charAt(0) === e.charAt(0).toLowerCase());
}
var pt = function(e) {
  return typeof e == "function" || typeof e == "object" && e !== null && !Array.isArray(e);
}, Cn = function(e) {
  return e !== "__proto__" && e !== "constructor" && e !== "prototype";
};
function En(e, n, t) {
  var o = e[t];
  pt(n) && pt(o) ? rr(o, n) : e[t] = n;
}
function rr(e) {
  for (var n = arguments.length, t = new Array(n > 1 ? n - 1 : 0), o = 1; o < n; o++)
    t[o - 1] = arguments[o];
  for (var a = 0, c = t; a < c.length; a++) {
    var s = c[a];
    if (pt(s))
      for (var f in s)
        Cn(f) && En(e, s[f], f);
  }
  return e;
}
var nr = Ye.createContext();
nr.Consumer;
var st = {};
function or(e, n, t) {
  var o = yt(e), a = !Ze(e), c = n.attrs, s = c === void 0 ? Qe : c, f = n.componentId, h = f === void 0 ? function(y, T) {
    var _ = typeof y != "string" ? "sc" : it(y);
    st[_] = (st[_] || 0) + 1;
    var d = _ + "-" + Sn("5.3.6" + _ + st[_]);
    return T ? T + "-" + d : d;
  }(n.displayName, n.parentComponentId) : f, v = n.displayName, g = v === void 0 ? function(y) {
    return Ze(y) ? "styled." + y : "Styled(" + ut(y) + ")";
  }(e) : v, C = n.displayName && n.componentId ? it(n.displayName) + "-" + n.componentId : n.componentId || h, D = o && e.attrs ? Array.prototype.concat(e.attrs, s).filter(Boolean) : s, F = n.shouldForwardProp;
  o && e.shouldForwardProp && (F = n.shouldForwardProp ? function(y, T, _) {
    return e.shouldForwardProp(y, T, _) && n.shouldForwardProp(y, T, _);
  } : e.shouldForwardProp);
  var E, O = new nn(t, C, o ? e.componentStyle : void 0), x = O.isStatic && s.length === 0, I = function(y, T) {
    return function(_, d, ae, he) {
      var ce = _.attrs, ue = _.componentStyle, me = _.defaultProps, we = _.foldedComponentIds, ie = _.shouldForwardProp, G = _.styledComponentId, te = _.target;
      process.env.NODE_ENV !== "production" && Ot(G);
      var K = function(A, i, j) {
        A === void 0 && (A = Pe);
        var p = Ae({}, i, { theme: A }), U = {};
        return j.forEach(function(M) {
          var $, l, X, Z = M;
          for ($ in Ge(Z) && (Z = Z(p)), Z)
            p[$] = U[$] = $ === "className" ? (l = U[$], X = Z[$], l && X ? l + " " + X : l || X) : Z[$];
        }), [p, U];
      }(yn(d, ye(nr), me) || Pe, d, ce), be = K[0], le = K[1], se = function(A, i, j, p) {
        var U = ln(), M = un(), $ = i ? A.generateAndInjectStyles(Pe, U, M) : A.generateAndInjectStyles(j, U, M);
        return process.env.NODE_ENV !== "production" && Ot($), process.env.NODE_ENV !== "production" && !i && p && p($), $;
      }(ue, he, be, process.env.NODE_ENV !== "production" ? _.warnTooManyClasses : void 0), Se = ae, ve = le.$as || d.$as || le.as || d.as || te, Ce = Ze(ve), S = le !== d ? Ae({}, d, {}, le) : d, w = {};
      for (var b in S)
        b[0] !== "$" && b !== "as" && (b === "forwardedAs" ? w.as = S[b] : (ie ? ie(b, Dt, ve) : !Ce || Dt(b)) && (w[b] = S[b]));
      return d.style && le.style !== d.style && (w.style = Ae({}, d.style, {}, le.style)), w.className = Array.prototype.concat(we, G, se !== G ? se : null, d.className, le.className).filter(Boolean).join(" "), w.ref = Se, Cr(ve, w);
    }(E, y, T, x);
  };
  return I.displayName = g, (E = Ye.forwardRef(I)).attrs = D, E.componentStyle = O, E.displayName = g, E.shouldForwardProp = F, E.foldedComponentIds = o ? Array.prototype.concat(e.foldedComponentIds, e.styledComponentId) : Qe, E.styledComponentId = C, E.target = o ? e.target : e, E.withComponent = function(y) {
    var T = n.componentId, _ = function(ae, he) {
      if (ae == null)
        return {};
      var ce, ue, me = {}, we = Object.keys(ae);
      for (ue = 0; ue < we.length; ue++)
        ce = we[ue], he.indexOf(ce) >= 0 || (me[ce] = ae[ce]);
      return me;
    }(n, ["componentId"]), d = T && T + "-" + (Ze(y) ? y : it(ut(y)));
    return or(y, Ae({}, _, { attrs: D, componentId: d }), t);
  }, Object.defineProperty(E, "defaultProps", { get: function() {
    return this._foldedDefaultProps;
  }, set: function(y) {
    this._foldedDefaultProps = o ? rr({}, e.defaultProps, y) : y;
  } }), process.env.NODE_ENV !== "production" && (gn(g, C), E.warnTooManyClasses = function(y, T) {
    var _ = {}, d = !1;
    return function(ae) {
      if (!d && (_[ae] = !0, Object.keys(_).length >= 200)) {
        var he = T ? ' with the id of "' + T + '"' : "";
        console.warn("Over 200 classes were generated for component " + y + he + `.
Consider using the attrs method, together with a style object for frequently changed styles.
Example:
  const Component = styled.div.attrs(props => ({
    style: {
      background: props.background,
    },
  }))\`width: 100%;\`

  <Component />`), d = !0, _ = {};
      }
    };
  }(g, C)), E.toString = function() {
    return "." + E.styledComponentId;
  }, a && $r(E, e, { attrs: !0, componentStyle: !0, displayName: !0, foldedComponentIds: !0, shouldForwardProp: !0, styledComponentId: !0, target: !0, withComponent: !0 }), E;
}
var ht = function(e) {
  return function n(t, o, a) {
    if (a === void 0 && (a = Pe), !qe.exports.isValidElementType(o))
      return Be(1, String(o));
    var c = function() {
      return t(o, a, vn.apply(void 0, arguments));
    };
    return c.withConfig = function(s) {
      return n(t, o, Ae({}, a, {}, s));
    }, c.attrs = function(s) {
      return n(t, o, Ae({}, a, { attrs: Array.prototype.concat(a.attrs, s).filter(Boolean) }));
    }, c;
  }(or, e);
};
["a", "abbr", "address", "area", "article", "aside", "audio", "b", "base", "bdi", "bdo", "big", "blockquote", "body", "br", "button", "canvas", "caption", "cite", "code", "col", "colgroup", "data", "datalist", "dd", "del", "details", "dfn", "dialog", "div", "dl", "dt", "em", "embed", "fieldset", "figcaption", "figure", "footer", "form", "h1", "h2", "h3", "h4", "h5", "h6", "head", "header", "hgroup", "hr", "html", "i", "iframe", "img", "input", "ins", "kbd", "keygen", "label", "legend", "li", "link", "main", "map", "mark", "marquee", "menu", "menuitem", "meta", "meter", "nav", "noscript", "object", "ol", "optgroup", "option", "output", "p", "param", "picture", "pre", "progress", "q", "rp", "rt", "ruby", "s", "samp", "script", "section", "select", "small", "source", "span", "strong", "style", "sub", "summary", "sup", "table", "tbody", "td", "textarea", "tfoot", "th", "thead", "time", "title", "tr", "track", "u", "ul", "var", "video", "wbr", "circle", "clipPath", "defs", "ellipse", "foreignObject", "g", "image", "line", "linearGradient", "marker", "mask", "path", "pattern", "polygon", "polyline", "radialGradient", "rect", "stop", "svg", "text", "textPath", "tspan"].forEach(function(e) {
  ht[e] = ht(e);
});
process.env.NODE_ENV !== "production" && typeof navigator < "u" && navigator.product === "ReactNative" && console.warn(`It looks like you've imported 'styled-components' on React Native.
Perhaps you're looking to import 'styled-components/native'?
Read more about this at https://www.styled-components.com/docs/basics#react-native`), process.env.NODE_ENV !== "production" && process.env.NODE_ENV !== "test" && typeof window < "u" && (window["__styled-components-init__"] = window["__styled-components-init__"] || 0, window["__styled-components-init__"] === 1 && console.warn(`It looks like there are several instances of 'styled-components' initialized in this application. This may cause dynamic styles to not render properly, errors during the rehydration process, a missing theme prop, and makes your application bigger without good reason.

See https://s-c.sh/2BAXzed for more info.`), window["__styled-components-init__"] += 1);
const fe = ht, Ne = {
  walletName: "",
  walletInstance: {}
}, _n = () => {
  const [e, n] = ee(void 0), [t, o] = ee(!1), [a, c] = ee(Ne.walletInstance), [s, f] = ee(Ne.walletName), h = ct(async (g) => {
    o(!0);
    try {
      const C = await qt.enable(g);
      c(C), f(g), n(void 0);
    } catch (C) {
      n(C), console.error(C);
    }
    o(!1);
  }, []), v = ct(() => {
    f(Ne.walletName), c(Ne.walletInstance);
  }, []);
  return {
    hasConnectedWallet: Ne.walletName !== s,
    connectedWalletInstance: a,
    connectedWalletName: s,
    connectingWallet: t,
    connectWallet: h,
    disconnect: v,
    error: e
  };
}, Te = Er({
  hasConnectedWallet: !1,
  connectedWalletInstance: Ne.walletInstance,
  connectedWalletName: Ne.walletName,
  connectingWallet: !1
});
var tt = { exports: {} }, Ve = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ut;
function xn() {
  if (Ut)
    return Ve;
  Ut = 1;
  var e = Ye, n = Symbol.for("react.element"), t = Symbol.for("react.fragment"), o = Object.prototype.hasOwnProperty, a = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, c = { key: !0, ref: !0, __self: !0, __source: !0 };
  function s(f, h, v) {
    var g, C = {}, D = null, F = null;
    v !== void 0 && (D = "" + v), h.key !== void 0 && (D = "" + h.key), h.ref !== void 0 && (F = h.ref);
    for (g in h)
      o.call(h, g) && !c.hasOwnProperty(g) && (C[g] = h[g]);
    if (f && f.defaultProps)
      for (g in h = f.defaultProps, h)
        C[g] === void 0 && (C[g] = h[g]);
    return { $$typeof: n, type: f, key: D, ref: F, props: C, _owner: a.current };
  }
  return Ve.Fragment = t, Ve.jsx = s, Ve.jsxs = s, Ve;
}
var Ue = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ht;
function Rn() {
  return Ht || (Ht = 1, process.env.NODE_ENV !== "production" && function() {
    var e = Ye, n = Symbol.for("react.element"), t = Symbol.for("react.portal"), o = Symbol.for("react.fragment"), a = Symbol.for("react.strict_mode"), c = Symbol.for("react.profiler"), s = Symbol.for("react.provider"), f = Symbol.for("react.context"), h = Symbol.for("react.forward_ref"), v = Symbol.for("react.suspense"), g = Symbol.for("react.suspense_list"), C = Symbol.for("react.memo"), D = Symbol.for("react.lazy"), F = Symbol.for("react.offscreen"), E = Symbol.iterator, O = "@@iterator";
    function x(r) {
      if (r === null || typeof r != "object")
        return null;
      var u = E && r[E] || r[O];
      return typeof u == "function" ? u : null;
    }
    var I = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function y(r) {
      {
        for (var u = arguments.length, m = new Array(u > 1 ? u - 1 : 0), k = 1; k < u; k++)
          m[k - 1] = arguments[k];
        T("error", r, m);
      }
    }
    function T(r, u, m) {
      {
        var k = I.ReactDebugCurrentFrame, Y = k.getStackAddendum();
        Y !== "" && (u += "%s", m = m.concat([Y]));
        var H = m.map(function(L) {
          return String(L);
        });
        H.unshift("Warning: " + u), Function.prototype.apply.call(console[r], console, H);
      }
    }
    var _ = !1, d = !1, ae = !1, he = !1, ce = !1, ue;
    ue = Symbol.for("react.module.reference");
    function me(r) {
      return !!(typeof r == "string" || typeof r == "function" || r === o || r === c || ce || r === a || r === v || r === g || he || r === F || _ || d || ae || typeof r == "object" && r !== null && (r.$$typeof === D || r.$$typeof === C || r.$$typeof === s || r.$$typeof === f || r.$$typeof === h || r.$$typeof === ue || r.getModuleId !== void 0));
    }
    function we(r, u, m) {
      var k = r.displayName;
      if (k)
        return k;
      var Y = u.displayName || u.name || "";
      return Y !== "" ? m + "(" + Y + ")" : m;
    }
    function ie(r) {
      return r.displayName || "Context";
    }
    function G(r) {
      if (r == null)
        return null;
      if (typeof r.tag == "number" && y("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof r == "function")
        return r.displayName || r.name || null;
      if (typeof r == "string")
        return r;
      switch (r) {
        case o:
          return "Fragment";
        case t:
          return "Portal";
        case c:
          return "Profiler";
        case a:
          return "StrictMode";
        case v:
          return "Suspense";
        case g:
          return "SuspenseList";
      }
      if (typeof r == "object")
        switch (r.$$typeof) {
          case f:
            var u = r;
            return ie(u) + ".Consumer";
          case s:
            var m = r;
            return ie(m._context) + ".Provider";
          case h:
            return we(r, r.render, "ForwardRef");
          case C:
            var k = r.displayName || null;
            return k !== null ? k : G(r.type) || "Memo";
          case D: {
            var Y = r, H = Y._payload, L = Y._init;
            try {
              return G(L(H));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var te = Object.assign, K = 0, be, le, se, Se, ve, Ce, S;
    function w() {
    }
    w.__reactDisabledLog = !0;
    function b() {
      {
        if (K === 0) {
          be = console.log, le = console.info, se = console.warn, Se = console.error, ve = console.group, Ce = console.groupCollapsed, S = console.groupEnd;
          var r = {
            configurable: !0,
            enumerable: !0,
            value: w,
            writable: !0
          };
          Object.defineProperties(console, {
            info: r,
            log: r,
            warn: r,
            error: r,
            group: r,
            groupCollapsed: r,
            groupEnd: r
          });
        }
        K++;
      }
    }
    function A() {
      {
        if (K--, K === 0) {
          var r = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: te({}, r, {
              value: be
            }),
            info: te({}, r, {
              value: le
            }),
            warn: te({}, r, {
              value: se
            }),
            error: te({}, r, {
              value: Se
            }),
            group: te({}, r, {
              value: ve
            }),
            groupCollapsed: te({}, r, {
              value: Ce
            }),
            groupEnd: te({}, r, {
              value: S
            })
          });
        }
        K < 0 && y("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var i = I.ReactCurrentDispatcher, j;
    function p(r, u, m) {
      {
        if (j === void 0)
          try {
            throw Error();
          } catch (Y) {
            var k = Y.stack.trim().match(/\n( *(at )?)/);
            j = k && k[1] || "";
          }
        return `
` + j + r;
      }
    }
    var U = !1, M;
    {
      var $ = typeof WeakMap == "function" ? WeakMap : Map;
      M = new $();
    }
    function l(r, u) {
      if (!r || U)
        return "";
      {
        var m = M.get(r);
        if (m !== void 0)
          return m;
      }
      var k;
      U = !0;
      var Y = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var H;
      H = i.current, i.current = null, b();
      try {
        if (u) {
          var L = function() {
            throw Error();
          };
          if (Object.defineProperty(L.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(L, []);
            } catch (ke) {
              k = ke;
            }
            Reflect.construct(r, [], L);
          } else {
            try {
              L.call();
            } catch (ke) {
              k = ke;
            }
            r.call(L.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (ke) {
            k = ke;
          }
          r();
        }
      } catch (ke) {
        if (ke && k && typeof ke.stack == "string") {
          for (var W = ke.stack.split(`
`), de = k.stack.split(`
`), Q = W.length - 1, ne = de.length - 1; Q >= 1 && ne >= 0 && W[Q] !== de[ne]; )
            ne--;
          for (; Q >= 1 && ne >= 0; Q--, ne--)
            if (W[Q] !== de[ne]) {
              if (Q !== 1 || ne !== 1)
                do
                  if (Q--, ne--, ne < 0 || W[Q] !== de[ne]) {
                    var ge = `
` + W[Q].replace(" at new ", " at ");
                    return r.displayName && ge.includes("<anonymous>") && (ge = ge.replace("<anonymous>", r.displayName)), typeof r == "function" && M.set(r, ge), ge;
                  }
                while (Q >= 1 && ne >= 0);
              break;
            }
        }
      } finally {
        U = !1, i.current = H, A(), Error.prepareStackTrace = Y;
      }
      var Le = r ? r.displayName || r.name : "", Pt = Le ? p(Le) : "";
      return typeof r == "function" && M.set(r, Pt), Pt;
    }
    function X(r, u, m) {
      return l(r, !1);
    }
    function Z(r) {
      var u = r.prototype;
      return !!(u && u.isReactComponent);
    }
    function P(r, u, m) {
      if (r == null)
        return "";
      if (typeof r == "function")
        return l(r, Z(r));
      if (typeof r == "string")
        return p(r);
      switch (r) {
        case v:
          return p("Suspense");
        case g:
          return p("SuspenseList");
      }
      if (typeof r == "object")
        switch (r.$$typeof) {
          case h:
            return X(r.render);
          case C:
            return P(r.type, u, m);
          case D: {
            var k = r, Y = k._payload, H = k._init;
            try {
              return P(H(Y), u, m);
            } catch {
            }
          }
        }
      return "";
    }
    var J = Object.prototype.hasOwnProperty, z = {}, re = I.ReactDebugCurrentFrame;
    function Ee(r) {
      if (r) {
        var u = r._owner, m = P(r.type, r._source, u ? u.type : null);
        re.setExtraStackFrame(m);
      } else
        re.setExtraStackFrame(null);
    }
    function oe(r, u, m, k, Y) {
      {
        var H = Function.call.bind(J);
        for (var L in r)
          if (H(r, L)) {
            var W = void 0;
            try {
              if (typeof r[L] != "function") {
                var de = Error((k || "React class") + ": " + m + " type `" + L + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof r[L] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw de.name = "Invariant Violation", de;
              }
              W = r[L](u, L, k, m, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (Q) {
              W = Q;
            }
            W && !(W instanceof Error) && (Ee(Y), y("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", k || "React class", m, L, typeof W), Ee(null)), W instanceof Error && !(W.message in z) && (z[W.message] = !0, Ee(Y), y("Failed %s type: %s", m, W.message), Ee(null));
          }
      }
    }
    var je = Array.isArray;
    function xe(r) {
      return je(r);
    }
    function pe(r) {
      {
        var u = typeof Symbol == "function" && Symbol.toStringTag, m = u && r[Symbol.toStringTag] || r.constructor.name || "Object";
        return m;
      }
    }
    function R(r) {
      try {
        return q(r), !1;
      } catch {
        return !0;
      }
    }
    function q(r) {
      return "" + r;
    }
    function We(r) {
      if (R(r))
        return y("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", pe(r)), q(r);
    }
    var Re = I.ReactCurrentOwner, _e = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Ct, Et, rt;
    rt = {};
    function ir(r) {
      if (J.call(r, "ref")) {
        var u = Object.getOwnPropertyDescriptor(r, "ref").get;
        if (u && u.isReactWarning)
          return !1;
      }
      return r.ref !== void 0;
    }
    function sr(r) {
      if (J.call(r, "key")) {
        var u = Object.getOwnPropertyDescriptor(r, "key").get;
        if (u && u.isReactWarning)
          return !1;
      }
      return r.key !== void 0;
    }
    function cr(r, u) {
      if (typeof r.ref == "string" && Re.current && u && Re.current.stateNode !== u) {
        var m = G(Re.current.type);
        rt[m] || (y('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', G(Re.current.type), r.ref), rt[m] = !0);
      }
    }
    function lr(r, u) {
      {
        var m = function() {
          Ct || (Ct = !0, y("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", u));
        };
        m.isReactWarning = !0, Object.defineProperty(r, "key", {
          get: m,
          configurable: !0
        });
      }
    }
    function ur(r, u) {
      {
        var m = function() {
          Et || (Et = !0, y("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", u));
        };
        m.isReactWarning = !0, Object.defineProperty(r, "ref", {
          get: m,
          configurable: !0
        });
      }
    }
    var dr = function(r, u, m, k, Y, H, L) {
      var W = {
        $$typeof: n,
        type: r,
        key: u,
        ref: m,
        props: L,
        _owner: H
      };
      return W._store = {}, Object.defineProperty(W._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(W, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: k
      }), Object.defineProperty(W, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: Y
      }), Object.freeze && (Object.freeze(W.props), Object.freeze(W)), W;
    };
    function fr(r, u, m, k, Y) {
      {
        var H, L = {}, W = null, de = null;
        m !== void 0 && (We(m), W = "" + m), sr(u) && (We(u.key), W = "" + u.key), ir(u) && (de = u.ref, cr(u, Y));
        for (H in u)
          J.call(u, H) && !_e.hasOwnProperty(H) && (L[H] = u[H]);
        if (r && r.defaultProps) {
          var Q = r.defaultProps;
          for (H in Q)
            L[H] === void 0 && (L[H] = Q[H]);
        }
        if (W || de) {
          var ne = typeof r == "function" ? r.displayName || r.name || "Unknown" : r;
          W && lr(L, ne), de && ur(L, ne);
        }
        return dr(r, W, de, Y, k, Re.current, L);
      }
    }
    var nt = I.ReactCurrentOwner, _t = I.ReactDebugCurrentFrame;
    function $e(r) {
      if (r) {
        var u = r._owner, m = P(r.type, r._source, u ? u.type : null);
        _t.setExtraStackFrame(m);
      } else
        _t.setExtraStackFrame(null);
    }
    var ot;
    ot = !1;
    function at(r) {
      return typeof r == "object" && r !== null && r.$$typeof === n;
    }
    function xt() {
      {
        if (nt.current) {
          var r = G(nt.current.type);
          if (r)
            return `

Check the render method of \`` + r + "`.";
        }
        return "";
      }
    }
    function pr(r) {
      {
        if (r !== void 0) {
          var u = r.fileName.replace(/^.*[\\\/]/, ""), m = r.lineNumber;
          return `

Check your code at ` + u + ":" + m + ".";
        }
        return "";
      }
    }
    var Rt = {};
    function hr(r) {
      {
        var u = xt();
        if (!u) {
          var m = typeof r == "string" ? r : r.displayName || r.name;
          m && (u = `

Check the top-level render call using <` + m + ">.");
        }
        return u;
      }
    }
    function kt(r, u) {
      {
        if (!r._store || r._store.validated || r.key != null)
          return;
        r._store.validated = !0;
        var m = hr(u);
        if (Rt[m])
          return;
        Rt[m] = !0;
        var k = "";
        r && r._owner && r._owner !== nt.current && (k = " It was passed a child from " + G(r._owner.type) + "."), $e(r), y('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', m, k), $e(null);
      }
    }
    function At(r, u) {
      {
        if (typeof r != "object")
          return;
        if (xe(r))
          for (var m = 0; m < r.length; m++) {
            var k = r[m];
            at(k) && kt(k, u);
          }
        else if (at(r))
          r._store && (r._store.validated = !0);
        else if (r) {
          var Y = x(r);
          if (typeof Y == "function" && Y !== r.entries)
            for (var H = Y.call(r), L; !(L = H.next()).done; )
              at(L.value) && kt(L.value, u);
        }
      }
    }
    function mr(r) {
      {
        var u = r.type;
        if (u == null || typeof u == "string")
          return;
        var m;
        if (typeof u == "function")
          m = u.propTypes;
        else if (typeof u == "object" && (u.$$typeof === h || u.$$typeof === C))
          m = u.propTypes;
        else
          return;
        if (m) {
          var k = G(u);
          oe(m, r.props, "prop", k, r);
        } else if (u.PropTypes !== void 0 && !ot) {
          ot = !0;
          var Y = G(u);
          y("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", Y || "Unknown");
        }
        typeof u.getDefaultProps == "function" && !u.getDefaultProps.isReactClassApproved && y("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function vr(r) {
      {
        for (var u = Object.keys(r.props), m = 0; m < u.length; m++) {
          var k = u[m];
          if (k !== "children" && k !== "key") {
            $e(r), y("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", k), $e(null);
            break;
          }
        }
        r.ref !== null && ($e(r), y("Invalid attribute `ref` supplied to `React.Fragment`."), $e(null));
      }
    }
    function Tt(r, u, m, k, Y, H) {
      {
        var L = me(r);
        if (!L) {
          var W = "";
          (r === void 0 || typeof r == "object" && r !== null && Object.keys(r).length === 0) && (W += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var de = pr(Y);
          de ? W += de : W += xt();
          var Q;
          r === null ? Q = "null" : xe(r) ? Q = "array" : r !== void 0 && r.$$typeof === n ? (Q = "<" + (G(r.type) || "Unknown") + " />", W = " Did you accidentally export a JSX literal instead of a component?") : Q = typeof r, y("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", Q, W);
        }
        var ne = fr(r, u, m, Y, H);
        if (ne == null)
          return ne;
        if (L) {
          var ge = u.children;
          if (ge !== void 0)
            if (k)
              if (xe(ge)) {
                for (var Le = 0; Le < ge.length; Le++)
                  At(ge[Le], r);
                Object.freeze && Object.freeze(ge);
              } else
                y("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              At(ge, r);
        }
        return r === o ? vr(ne) : mr(ne), ne;
      }
    }
    function gr(r, u, m) {
      return Tt(r, u, m, !0);
    }
    function yr(r, u, m) {
      return Tt(r, u, m, !1);
    }
    var wr = yr, br = gr;
    Ue.Fragment = o, Ue.jsx = wr, Ue.jsxs = br;
  }()), Ue;
}
(function(e) {
  process.env.NODE_ENV === "production" ? e.exports = xn() : e.exports = Rn();
})(tt);
const De = tt.exports.Fragment, N = tt.exports.jsx, Me = tt.exports.jsxs, Xn = ({
  children: e
}) => {
  const n = _n();
  return /* @__PURE__ */ N(Te.Provider, {
    value: n,
    children: e
  });
}, Kn = (e = 0) => {
  const [n, t] = ee(), {
    hasConnectedWallet: o,
    connectedWalletName: a,
    connectedWalletInstance: c
  } = ye(Te);
  return Oe(() => {
    o && c.getUsedAddresses().then((s) => {
      s[e] && t(s[e]);
    });
  }, [e, a]), n;
}, Zn = () => {
  const [e, n] = ee(), {
    hasConnectedWallet: t,
    connectedWalletName: o,
    connectedWalletInstance: a
  } = ye(Te);
  return Oe(() => {
    t && a.getAssets().then(n);
  }, [o]), e;
}, bt = () => {
  const [e, n] = ee([]);
  return Oe(() => {
    n(qt.getInstalledWallets());
  }, []), e;
}, kn = () => {
  const [e, n] = ee(), {
    hasConnectedWallet: t,
    connectedWalletName: o,
    connectedWalletInstance: a
  } = ye(Te);
  return Oe(() => {
    t && a.getLovelace().then(n);
  }, [o]), e;
}, Jn = () => {
  const [e, n] = ee(), {
    hasConnectedWallet: t,
    connectedWalletName: o,
    connectedWalletInstance: a
  } = ye(Te);
  return Oe(() => {
    t && a.getNetworkId().then(n);
  }, [o]), e;
}, An = (e = 0) => {
  const [n, t] = ee(), {
    hasConnectedWallet: o,
    connectedWalletName: a,
    connectedWalletInstance: c
  } = ye(Te);
  return Oe(() => {
    o && c.getRewardAddresses().then((s) => {
      s[e] && t(s[e]);
    });
  }, [e, a]), n;
}, St = () => {
  const {
    hasConnectedWallet: e,
    connectedWalletName: n,
    connectedWalletInstance: t,
    connectingWallet: o,
    connectWallet: a,
    disconnect: c,
    error: s
  } = ye(Te);
  if (a === void 0 || c === void 0)
    throw new Error("Can't call useWallet outside of the WalletProvider context");
  return {
    name: n,
    connecting: o,
    connected: e,
    wallet: t,
    connect: a,
    disconnect: c,
    error: s
  };
}, Qn = () => {
  const [e, n] = ee(), [t, o] = ee(), [a, c] = ee(!1), {
    hasConnectedWallet: s,
    connectedWalletInstance: f
  } = ye(Te), h = ct(async (v) => {
    c(!0);
    try {
      if (s) {
        const g = await f.submitTx(v);
        n(void 0), o(g);
      }
      throw new Error("Please make sure to connect a wallet before calling useWalletSubmit");
    } catch (g) {
      n(g), console.error(g);
    }
    c(!1);
  }, []);
  return {
    error: e,
    result: t,
    submitting: a,
    submitTx: h
  };
}, Tn = (e = {}) => {
  const {
    era: n,
    parameters: t
  } = e, {
    hasConnectedWallet: o,
    connectedWalletInstance: a
  } = ye(Te), [c] = ee(() => {
    if (o)
      return new _r({
        initiator: a,
        parameters: t,
        era: n
      });
    throw new Error("Please make sure to connect a wallet before calling useWalletTx");
  });
  return c;
}, Pn = fe.svg.withConfig({
  displayName: "CheckMark__StyledCheckMark",
  componentId: "sc-bdkad9-0"
})({
  margin: "0.5rem",
  height: "1.5rem"
}), On = () => /* @__PURE__ */ N(Pn, {
  fill: "currentColor",
  viewBox: "0 0 512 512",
  xmlns: "http://www.w3.org/2000/svg",
  children: /* @__PURE__ */ N("path", {
    d: "M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
  })
}), In = fe.div.withConfig({
  displayName: "MenuItem__StyledItem",
  componentId: "sc-obisy3-0"
})({
  display: "flex",
  cursor: "pointer",
  alignItems: "center",
  paddingLeft: "1rem",
  paddingRight: "1rem",
  paddingTop: "0.5rem",
  paddingBottom: "0.5rem",
  opacity: "0.8",
  ":hover": {
    opacity: "1"
  }
}), Nn = fe.img.withConfig({
  displayName: "MenuItem__StyledIcon",
  componentId: "sc-obisy3-1"
})({
  margin: "0.25rem",
  height: "2rem",
  paddingRight: "0.5rem"
}), Dn = fe.span.withConfig({
  displayName: "MenuItem__StyledName",
  componentId: "sc-obisy3-2"
})({
  fontSize: "1.25rem",
  lineHeight: "1.75rem",
  fontWeight: "400"
}), Mn = fe.span.withConfig({
  displayName: "MenuItem__StyledActive",
  componentId: "sc-obisy3-3"
})({
  marginLeft: "auto"
}), mt = ({
  icon: e,
  label: n,
  action: t,
  active: o
}) => /* @__PURE__ */ Me(In, {
  onClick: t,
  children: [e && /* @__PURE__ */ N(Nn, {
    src: e
  }), /* @__PURE__ */ N(Dn, {
    className: "mr-menu-item",
    children: n.split(" ").map((a) => a.charAt(0).toUpperCase() + a.slice(1).toLowerCase()).join(" ")
  }), o && /* @__PURE__ */ N(Mn, {
    children: /* @__PURE__ */ N(On, {})
  })]
}), jn = fe.svg.withConfig({
  displayName: "ChevronDown__StyledChevronDown",
  componentId: "sc-17n3mfh-0"
})({
  margin: "0.5rem",
  height: "1.5rem"
}), ar = () => /* @__PURE__ */ N(jn, {
  fill: "none",
  "aria-hidden": "true",
  viewBox: "0 0 24 24",
  stroke: "currentColor",
  xmlns: "http://www.w3.org/2000/svg",
  children: /* @__PURE__ */ N("path", {
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: "2",
    d: "M19 9l-7 7-7-7"
  })
}), Gt = fe.img.withConfig({
  displayName: "WalletBalance__StyledIcon",
  componentId: "sc-8if913-0"
})({
  margin: "0.5rem",
  height: "1.5rem"
}), Wn = fe.span.withConfig({
  displayName: "WalletBalance__StyledSmall",
  componentId: "sc-8if913-1"
})({
  fontSize: "0.75rem",
  lineHeight: "1rem"
}), $n = ({
  connected: e,
  name: n,
  connecting: t,
  label: o
}) => {
  const a = bt().find((s) => s.name === n), c = kn();
  return e && c && a?.icon ? /* @__PURE__ */ Me(De, {
    children: [/* @__PURE__ */ N(Gt, {
      src: a.icon
    }), "\u20B3", " ", parseInt((parseInt(c, 10) / 1e6).toString(), 10), ".", /* @__PURE__ */ N(Wn, {
      children: c.substr(c.length - 6)
    })]
  }) : e && a?.icon ? /* @__PURE__ */ N(De, {
    children: /* @__PURE__ */ N(Gt, {
      src: a.icon
    })
  }) : t ? /* @__PURE__ */ N(De, {
    children: "Connecting..."
  }) : /* @__PURE__ */ Me(De, {
    children: [o, " ", /* @__PURE__ */ N(ar, {})]
  });
}, Ln = fe.button.withConfig({
  displayName: "CardanoWallet__StyledMenuButton",
  componentId: "sc-13vl5pq-0"
})({
  display: "flex",
  width: "15rem",
  alignItems: "center",
  justifyContent: "center",
  borderTopLeftRadius: "0.375rem",
  borderTopRightRadius: "0.375rem",
  borderWidth: "1px",
  paddingLeft: "1rem",
  paddingRight: "1rem",
  paddingTop: "0.5rem",
  paddingBottom: "0.5rem",
  fontSize: "1.125rem",
  lineHeight: "1.75rem",
  fontWeight: "400",
  "--tw-shadow": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
  "--tw-shadow-colored": "0 1px 2px 0 var(--tw-shadow-color)",
  boxShadow: "var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow)"
}), Fn = fe.div.withConfig({
  displayName: "CardanoWallet__StyledMenuList",
  componentId: "sc-13vl5pq-1"
})(({
  hidden: e
}) => [{
  position: "absolute",
  width: "15rem",
  borderBottomRightRadius: "0.375rem",
  borderBottomLeftRadius: "0.375rem",
  borderWidth: "1px",
  textAlign: "center",
  "--tw-shadow": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
  "--tw-shadow-colored": "0 1px 2px 0 var(--tw-shadow-color)",
  boxShadow: "var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow)",
  "--tw-backdrop-blur": "blur(8px)",
  backdropFilter: "var(--tw-backdrop-blur) var(--tw-backdrop-brightness) var(--tw-backdrop-contrast) var(--tw-backdrop-grayscale) var(--tw-backdrop-hue-rotate) var(--tw-backdrop-invert) var(--tw-backdrop-opacity) var(--tw-backdrop-saturate) var(--tw-backdrop-sepia)"
}, e && {
  display: "none"
}]), eo = ({
  label: e = "Connect Wallet",
  onConnected: n = void 0
}) => {
  const t = bt(), [o, a] = ee(!0), {
    connect: c,
    connecting: s,
    connected: f,
    disconnect: h,
    name: v
  } = St();
  return Oe(() => {
    f && n && n();
  }, [f]), /* @__PURE__ */ Me("div", {
    style: {
      width: "fit-content"
    },
    onMouseEnter: () => a(!1),
    onMouseLeave: () => a(!0),
    children: [/* @__PURE__ */ N(Ln, {
      type: "button",
      className: "mr-wallet-button",
      onClick: () => a(!o),
      children: /* @__PURE__ */ N($n, {
        name: v,
        connected: f,
        connecting: s,
        label: e
      })
    }), /* @__PURE__ */ N(Fn, {
      hidden: o,
      className: "mr-menu-list",
      children: !f && t.length > 0 ? t.map((g, C) => /* @__PURE__ */ N(mt, {
        icon: g.icon,
        label: g.name,
        action: () => {
          c(g.name), a(!o);
        },
        active: v === g.name
      }, C)) : t.length === 0 ? /* @__PURE__ */ N("span", {
        children: "No Wallet Found"
      }) : /* @__PURE__ */ N(De, {
        children: /* @__PURE__ */ N(mt, {
          active: !1,
          label: "disconnect",
          action: h,
          icon: void 0
        })
      })
    })]
  });
}, zn = fe.svg.withConfig({
  displayName: "MeshLogo__StyledLogo",
  componentId: "sc-1nwgent-0"
})({
  height: "4rem",
  padding: "0.5rem"
}), Yn = () => /* @__PURE__ */ N(zn, {
  fill: "currentColor",
  viewBox: "0 0 300 200",
  xmlns: "http://www.w3.org/2000/svg",
  children: /* @__PURE__ */ N("path", {
    d: "m289 127-45-60-45-60c-.9-1.3-2.4-2-4-2s-3.1.7-4 2l-37 49.3c-2 2.7-6 2.7-8 0l-37-49.3c-.9-1.3-2.4-2-4-2s-3.1.7-4 2l-45 60-45 60c-1.3 1.8-1.3 4.2 0 6l45 60c.9 1.3 2.4 2 4 2s3.1-.7 4-2l37-49.3c2-2.7 6-2.7 8 0l37 49.3c.9 1.3 2.4 2 4 2s3.1-.7 4-2l37-49.3c2-2.7 6-2.7 8 0l37 49.3c.9 1.3 2.4 2 4 2s3.1-.7 4-2l45-60c1.3-1.8 1.3-4.2 0-6zm-90-103.3 32.5 43.3c1.3 1.8 1.3 4.2 0 6l-32.5 43.3c-2 2.7-6 2.7-8 0l-32.5-43.3c-1.3-1.8-1.3-4.2 0-6l32.5-43.3c2-2.7 6-2.7 8 0zm-90 0 32.5 43.3c1.3 1.8 1.3 4.2 0 6l-32.5 43.3c-2 2.7-6 2.7-8 0l-32.5-43.3c-1.3-1.8-1.3-4.2 0-6l32.5-43.3c2-2.7 6-2.7 8 0zm-53 152.6-32.5-43.3c-1.3-1.8-1.3-4.2 0-6l32.5-43.3c2-2.7 6-2.7 8 0l32.5 43.3c1.3 1.8 1.3 4.2 0 6l-32.5 43.3c-2 2.7-6 2.7-8 0zm90 0-32.5-43.3c-1.3-1.8-1.3-4.2 0-6l32.5-43.3c2-2.7 6-2.7 8 0l32.5 43.3c1.3 1.8 1.3 4.2 0 6l-32.5 43.3c-2 2.7-6 2.7-8 0zm90 0-32.5-43.3c-1.3-1.8-1.3-4.2 0-6l32.5-43.3c2-2.7 6-2.7 8 0l32.5 43.3c1.3 1.8 1.3 4.2 0 6l-32.5 43.3c-2 2.7-6 2.7-8 0z"
  })
}), Bn = fe.a.withConfig({
  displayName: "MeshBadge__StyledLink",
  componentId: "sc-fcpuc3-0"
})({
  display: "flex",
  maxWidth: "fit-content",
  flexDirection: "column",
  alignItems: "center",
  borderRadius: "0.375rem",
  borderWidth: "1px",
  borderStyle: "solid",
  borderColor: "currentColor",
  padding: "0.25rem",
  fontSize: "1.25rem",
  lineHeight: "1.75rem",
  fontWeight: "600",
  "--tw-text-opacity": "1",
  color: "rgb(68 64 60 / var(--tw-text-opacity))",
  textDecorationLine: "none"
}), to = ({
  dark: e = !1
}) => /* @__PURE__ */ Me(Bn, {
  style: {
    color: e ? "#EEEEEE" : "#111111",
    backgroundColor: e ? "#111111" : "#EEEEEE"
  },
  href: "https://meshjs.dev/",
  rel: "noopener noreferrer",
  target: "_blank",
  children: [/* @__PURE__ */ N(Yn, {}), "Mesh"]
}), Vn = fe.button.withConfig({
  displayName: "StakeButton__StyledMenuButton",
  componentId: "sc-2hlzah-0"
})({
  display: "flex",
  width: "15rem",
  alignItems: "center",
  justifyContent: "center",
  borderTopLeftRadius: "0.375rem",
  borderTopRightRadius: "0.375rem",
  borderWidth: "1px",
  paddingLeft: "1rem",
  paddingRight: "1rem",
  paddingTop: "0.5rem",
  paddingBottom: "0.5rem",
  fontSize: "1.125rem",
  lineHeight: "1.75rem",
  fontWeight: "400",
  "--tw-shadow": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
  "--tw-shadow-colored": "0 1px 2px 0 var(--tw-shadow-color)",
  boxShadow: "var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow)"
}), Un = fe.div.withConfig({
  displayName: "StakeButton__StyledMenuList",
  componentId: "sc-2hlzah-1"
})(({
  hidden: e
}) => [{
  position: "absolute",
  width: "15rem",
  borderBottomRightRadius: "0.375rem",
  borderBottomLeftRadius: "0.375rem",
  borderWidth: "1px",
  textAlign: "center",
  "--tw-shadow": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
  "--tw-shadow-colored": "0 1px 2px 0 var(--tw-shadow-color)",
  boxShadow: "var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow)",
  "--tw-backdrop-blur": "blur(8px)",
  backdropFilter: "var(--tw-backdrop-blur) var(--tw-backdrop-brightness) var(--tw-backdrop-contrast) var(--tw-backdrop-grayscale) var(--tw-backdrop-hue-rotate) var(--tw-backdrop-invert) var(--tw-backdrop-opacity) var(--tw-backdrop-saturate) var(--tw-backdrop-sepia)"
}, e && {
  display: "none"
}]), ro = ({
  poolId: e,
  onCheck: n
}) => {
  const t = bt(), [o, a] = ee(!0), {
    connect: c,
    connecting: s,
    connected: f,
    name: h
  } = St();
  return /* @__PURE__ */ Me("div", {
    style: {
      width: "fit-content"
    },
    onMouseEnter: () => a(!1),
    onMouseLeave: () => a(!0),
    children: [/* @__PURE__ */ N(Vn, {
      type: "button",
      onClick: () => a(!o),
      children: f ? /* @__PURE__ */ N(Hn, {
        poolId: e,
        onCheck: n
      }) : s ? /* @__PURE__ */ N(De, {
        children: "Connecting..."
      }) : /* @__PURE__ */ Me(De, {
        children: ["Connect Wallet ", /* @__PURE__ */ N(ar, {})]
      })
    }), /* @__PURE__ */ N(Un, {
      hidden: o || f,
      children: t.length > 0 ? t.map((v, g) => /* @__PURE__ */ N(mt, {
        icon: v.icon,
        label: v.name,
        action: () => {
          c(v.name), a(!o);
        },
        active: h === v.name
      }, g)) : /* @__PURE__ */ N("span", {
        children: "No Wallet Found"
      })
    })]
  });
}, Hn = ({
  poolId: e,
  onCheck: n
}) => {
  const t = Tn(), {
    wallet: o
  } = St(), a = An(), [c, s] = ee(), [f, h] = ee(!1), [v, g] = ee(), [C, D] = ee(!1), [F, E] = ee(!1), O = async () => {
    try {
      if (h(!0), a) {
        const y = await n(a);
        g(y);
      }
      h(!1);
    } catch (y) {
      s(y);
    }
  }, x = async () => {
    D(!0), E(!1);
    try {
      if (a) {
        const y = await t.registerStake(a).delegateStake(a, e).build(), T = await o.signTx(y), _ = await o.submitTx(T);
        console.log("txHash", _), E(!0);
      }
    } catch (y) {
      console.error("error", y), s(y);
    }
    D(!1);
  }, I = async () => {
    D(!0), E(!1);
    try {
      if (a) {
        const y = await t.delegateStake(a, e).build(), T = await o.signTx(y), _ = await o.submitTx(T);
        E(!0);
      }
    } catch (y) {
      console.error("error", y), s(y);
    }
    D(!1);
  };
  return Oe(() => {
    O();
  }, [a]), f ? /* @__PURE__ */ N("span", {
    children: "Checking..."
  }) : C ? /* @__PURE__ */ N("span", {
    children: "Loading..."
  }) : F ? /* @__PURE__ */ N("span", {
    children: "Stake Delegated"
  }) : v?.active ? v.poolId === e ? /* @__PURE__ */ N("span", {
    children: "Stake Delegated"
  }) : /* @__PURE__ */ N("span", {
    onClick: I,
    children: "Delegate Stake"
  }) : /* @__PURE__ */ N("span", {
    onClick: x,
    children: "Register Address"
  });
};
export {
  eo as CardanoWallet,
  to as MeshBadge,
  Xn as MeshProvider,
  ro as StakeButton,
  Te as WalletContext,
  Kn as useAddress,
  Zn as useAssets,
  kn as useLovelace,
  Jn as useNetwork,
  An as useRewardAddress,
  St as useWallet,
  bt as useWalletList,
  Qn as useWalletSubmit,
  Tn as useWalletTx
};
